<h2>Description</h2>
<p>
  These tracks show the one-to-one v1_nfLO alignments of the T2T-CHM13 v2.0
  assembly to the GRCh38/hg38 and GRCh37/hg19 human reference assemblies.
</p>

<h2>Display Conventions</h2>
<p>
  The track displays boxes joined together by either single or double lines,
  with the boxes represent aligning regions, single lines indicating gaps that
  are largely due to a deletion in the CHM13 v2.0 assembly or an insertion in
  the GRCh38 or GRCh37, and double lines representing more complex gaps that
  involve substantial sequence in both assembly.
</p>


<h2>LiftOver chain file downloads</h2>
One-to-one liftOver chain files to and from GRCh38/hg38 and GRCh37/hg19 are available here:
<ul>
  <li> T2T CHM13 v2.0 to GRCh38/hg38 <a href="https://hgdownload.gi.ucsc.edu/hubs/GCA/009/914/755/GCA_009914755.4/liftOver/chm13v2-hg38.over.chain.gz">chm13v2-hg38.over.chain.gz</a>
  <li> T2T CHM13 v2.0 to GRCh37/hg19 <a href="https://hgdownload.gi.ucsc.edu/hubs/GCA/009/914/755/GCA_009914755.4/liftOver/chm13v2-hg19.over.chain.gz">chm13v2-hg19.over.chain.gz</a>
  <li> GRCh38/hg38 to T2T CHM13 v2.0 <a href="https://hgdownload.gi.ucsc.edu/hubs/GCA/009/914/755/GCA_009914755.4/liftOver/hg38-chm13v2.over.chain.gz">hg38-chm13v2.over.chain.gz</a>
  <li> GRCh37/hg19 to T2T CHM13 v2.0 <a href="https://hgdownload.gi.ucsc.edu/hubs/GCA/009/914/755/GCA_009914755.4/liftOver/hg19-chm13v2.over.chain.gz">hg19-chm13v2.over.chain.gz</a>
</ul>

The mask file for GRCh38/hg38
is <a href="https://hgdownload.gi.ucsc.edu/hubs/GCA/009/914/755/GCA_009914755.4/liftOver/hg38.liftover-mask.bed">hg38.liftover-mask.bed</a>.

<h2>Methods</h2>
<p>
<ul>
<h3>GRCh38/hg38 pre-processing</h3>
<p>
To prevent ambiguous alignments, all false duplications, as determined by the Genome in a Bottle Consortium
(<a href="https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/release/references/GRCh38/GCA_000001405.15_GRCh38_GRC_exclusions_T2Tv2.bed">GCA_000001405.15_GRCh38_GRC_exclusions_T2Tv2.bed</a>), as well as the GRCh38 <a href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=1285829831_crE6AIbRoN8GqiiHOPmjhYm2qTq2&db=hg38&c=chrX&g=centromeres">modeled centromeres</a>,
were masked from the GRCh38/hg38 primary assembly. In addition, unlocalized and unplaced (random) contigs were removed.
</p>

<h3>GRCh37/hg19 pre-processing</h3>
<p>
Unlocalized and unplaced (random) contigs were removed from the GRCh37/hg19 assembly.
</p>


<h3>Alignment and Chain Creation</h3>
<p>
For the minimap2-based pipeline, the initial chain file was generated using
<a href="https://github.com/evotools/nf-LO">nf-LO</a> v1.5.1 with
<a href="https://github.com/lh3/minimap2">minimap2</a> v2.24 alignments. These chains were then split at all locations that contained unaligned segments greater than 1kbp or gaps greater than 10kbp. Split chain files were then converted to PAF format with extended CIGAR strings using chaintools (https://doi.org/10.5281/zenodo.6342391, v0.1), and alignments between nonhomologous chromosomes were removed. The trim-paf operation of rustybam (https://zenodo.org/record/6342176, v0.1.29) was next used to remove overlapping alignments in the query sequence, and then the target sequence, to create 1:1 alignments. PAF alignments were converted back to the chain format with <a href="https://github.com/AndreaGuarracino/paf2chain">paf2chain</a> commit f68eeca, and finally, chaintools was used to generate the inverted chain file.
</p>

<p>
Full commands with parameters used were:
</p>
<pre><code>
    nextflow run main.nf --source GRCh38.fa --target chm13v2.0.fasta --outdir dir -profile local --aligner minimap2
    python chaintools/src/split.py -c input.chain -o input-split.chain
    python chaintools/src/to_paf.py -c input-split.chain -t target.fa -q query.fa -o input-split.paf
    awk '$1==$6' input-split.paf | rb break-paf --max-size 10000  | rb trim-paf -r | rb invert | rb trim-paf -r | rb invert > out.paf
    paf2chain -i out.paf > out.chain
    python chaintools/src/invert.py -c out.chain -o out_inverted.chain
</code></pre>

<p>
<a href="https://mrvollger.github.io/rustybam/#align-once">Rustybam trim-paf</a>
uses dynamic programming and the CIGAR string to find an optimal
splitting point between overlapping alignments in the query sequence. It
starts its trimming with the largest overlap and then recursively trims
smaller overlaps.
</p>

<p>
Results were validated by using chaintools to confirm that there were no
overlapping sequences with respect to both CHM13v2.0 and GRCh38 in the
released chain file. In addition, trimmed alignments were visually inspected
with SafFire to confirm their quality.
</p>
</ul>
</p>

<h2>Credits</h2>
<p>
The v1_nflo chains were generated by <a href="mailto:naechyun.chen@gmail.com">Nae-Chyun Chen&lt;naechyun.chen@gmail.com&gt;</a>
and <a href="mvollger@uw.edu">Mitchell Vollger&lt;mvollger@uw.edu&gt;</a>
</p>

<h2>References</h2>
<p>
<p>Nurk S, Koren S, Rhie A, Rautiainen M, et al. The complete sequence of a human genome. bioRxiv, 2021.</p>
</p>


