<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">ol{margin:0;padding:0}table td,table th{padding:0}.c0{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial";font-style:normal}.c4{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial";font-style:normal}.c1{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left}.c5{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;color:#1155cc;text-decoration:underline}.c6{background-color:#ffffff;max-width:468pt;padding:72pt 72pt 72pt 72pt}.c3{color:inherit;text-decoration:inherit}.c2{height:11pt}.title{padding-top:0pt;color:#000000;font-size:26pt;padding-bottom:3pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.subtitle{padding-top:0pt;color:#666666;font-size:15pt;padding-bottom:16pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}li{color:#000000;font-size:11pt;font-family:"Arial"}p{margin:0;color:#000000;font-size:11pt;font-family:"Arial"}h1{padding-top:20pt;color:#000000;font-size:20pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-size:16pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h3{padding-top:16pt;color:#434343;font-size:14pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h4{padding-top:14pt;color:#666666;font-size:12pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h5{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h6{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;font-style:italic;orphans:2;widows:2;text-align:left}</style></head><body class="c6"><p class="c1"><span class="c4">Description</span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span class="c0">This track represents the gene models for the Marmoset assembly, calJac4, generated using Comparative Annotation Toolkit (CAT). </span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span>CAT can be found on GitHub here: </span><span class="c5"><a class="c3" href="https://www.google.com/url?q=https://github.com/ComparativeGenomicsToolkit/Comparative-Annotation-Toolkit&amp;sa=D&amp;source=editors&amp;ust=1614125890361000&amp;usg=AOvVaw1f6vU6LJ331N95FQcBo39i">https://github.com/ComparativeGenomicsToolkit/Comparative-Annotation-Toolkit</a></span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span class="c4">Display convention</span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span class="c0">This track follows the display conventions for gene prediction tracks. The exons for putative non-coding genes and untranslated regions are represented by relatively thin blocks, while those for coding open reading frames are thicker. Gene names are displayed in &#39;pack&#39; or &#39;full&#39; mode. More information about each gene can be found by clicking on the specific gene/transcript model.</span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span class="c0">The following color key is used:</span></p><p class="c1"><span class="c0">Blue: protein coding</span></p><p class="c1"><span class="c0">Green: non-coding</span></p><p class="c1"><span class="c0">Purple: novel predictions from augCGP/augPB</span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span class="c4">Methods</span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span class="c0">Genome annotation for the calJac4 assembly was performed using Comparative Annotation Toolkit (CAT). CAT leverages whole-genome alignments generated by Cactus to transfer annotations from one source genome to one or more target genomes. CAT also runs AUGUSTUS in both the comparative gene prediction mode and in a single-genome mode that utilizes Iso-Seq data to predict alternative isoforms. CAT then combines all of these annotation methods into a final consensus annotation set that represents orthology relationships as well as species-specific information.</span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span class="c0">The genomes included in the alignment were human, bonobo, chimp, gorilla, orangutan, gibbon, marmoset, owl monkey, and rhesus macaque. The GENCODE v33 annotation for GRCh38 was used as a reference to map onto the calJac4 assembly. 20 IsoSeq libraries were included as evidence for the CAT annotation pipeline.</span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span class="c4">Contacts</span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span>Marina Haukness &lt;</span><span class="c5"><a class="c3" href="mailto:mhauknes@ucsc.edu">mhauknes@ucsc.edu</a></span><span class="c0">&gt; </span></p><p class="c1"><span>Mark Diekhans &lt;</span><span class="c5"><a class="c3" href="mailto:markd@ucsc.edu">markd@ucsc.edu</a></span><span class="c0">&gt; </span></p><p class="c1"><span>Ian Fiddes &lt;</span><span class="c5"><a class="c3" href="mailto:ian.t.fiddes@gmail.com">ian.t.fiddes@gmail.com</a></span><span class="c0">&gt;</span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span class="c4">References</span></p><p class="c1"><span class="c0">Fiddes IT, Armstrong J, Diekhans M, Nachtweide S, Kronenberg ZN, Underwood JG, Gordon D, Earl D, Keane T, Eichler EE et al. Comparative Annotation Toolkit (CAT)-simultaneous clade and personal genome annotation. Genome Res. 2018 Jul;28(7):1029-1038. PMID: 29884752; PMC: PMC6028123.</span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span class="c0">Stanke M, Diekhans M, Baertsch R, Haussler D. Using native and syntenically mapped cDNA alignments to improve de novo gene finding. Bioinformatics. 2008 Mar 1;24(5):637-44. PMID: 18218656.</span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span class="c0">Stanke M, Steinkamp R, Waack S, Morgenstern B. AUGUSTUS: a web server for gene finding in eukaryotes. Nucleic Acids Res. 2004 Jul 1;32(Web Server issue):W309-12. PMID: 15215400; PMC: PMC441517.</span></p><p class="c1 c2"><span class="c0"></span></p><p class="c1"><span class="c0">Armstrong J, Hickey G, Diekhans M, Fiddes I.T, Novak A.M, Deran A, Fang Q, Xie D, Feng S, Stiller J, Genereux D, Johnson J, Marinescu V.D, Alf&ouml;ldi J, Harris R.S, Lindblad-Toh K, Haussler D, Karlsson E, Jarvis E.D, Zhang G, Paten B. Progressive Cactus is a multiple-genome aligner for the thousand-genome era. Nature. 2020 Nov 11; 587(7833):246- 251. PMID: 33177663 PMC: PMC7673649</span></p><p class="c1 c2"><span class="c0"></span></p></body></html>