<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser Wiggle Tracks" -->
<!--#set var="ROOT" value="../.." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<!-- Graph-Based Track Help Page -->

<a name="Topic5"></a><br>
<h1>Configuring graph-based tracks</h1> 
<p> 
Genome Browser graph-based tracks may be configured in a variety of ways to highlight different 
aspects of the displayed information. Some track types, such as BAM tracks and certain gene tracks, 
offer dynamic calculation of items in the current window to display a density graph where the height
is proportional to the number of items mapped to each genomic position. The following information 
explains each of the available configuration options.</p> 
<p> 
<strong>Overlay method</strong>: The default &quot;transparent&quot; setting displays the colored 
transparent graphs of multiple subtracks overlayed in the same vertical space. The 
&quot;solid&quot; setting displays the colored opaque graphs of multiple subtracks overlayed in 
the same vertical space.  The &quot;stacked&quot; setting displays each graph stacked on top of 
each other where the high point of the graph is the sum of all the values. The &quot;none&quot; 
setting displays each graph in its own vertical space. Note that not all graph-based tracks 
include this option.</p>
<p> 
<strong>Type of graph</strong>: The default &quot;Bar&quot; setting depicts the graph data using 
color-filled bars. To view the data as a series of points or lines, select the 
&quot;Points&quot; setting.</p>  
<p> 
<strong>Track height</strong>: To change the default display height of the graph in pixels, type 
in a value from within the indicated range.</p>  
<p> 
<strong>Vertical viewing range</strong>: The min and max values specify the vertical portion of 
the graph that is displayed (default range is 30-70). These numbers can be used to set data 
threshold indicators. For example, to display only those GC values greater than 50 percent, set 
the min value to &quot;50&quot;.</p> 
<p> 
<strong>Data view scaling</strong>: The default &quot;Use vertical viewing range setting&quot; 
option displays the data using the parameters specified in the Vertical viewing range setting. To 
configure the graph to automatically scale to a range defined by the minimum and maximum data 
points in the current view, select the &quot;Auto-scale to data view&quot; option. To keep the 
y=0 value in view at all times when Auto-scale is selected, set &quot;Always include 
zero&quot; to &quot;ON&quot;.</p>  
<p> 
<strong>Transform function</strong>: Transforms the data points by the function selected in the 
drop-down menu. Usually the default setting is &quot;None&quot;.</p> 
<p> 
<strong>Windowing function</strong>: When a  view is too large to show individual data values, the
values must be combined to produce a plot point. This option specifies the combining function to 
be used (default is &quot;Mean&quot;):</p> 
<ul> 
  <li> 
  &quot;Mean+whiskers&quot; - displays the mean in a dark shade, one standard deviation around the
  mean in a medium shade, and the maximum/minimum in a light shade. For bar graphs only the mean, 
  the mean plus a standard deviation, and the max are visible. This mode is not available if the 
  Overlay method is stacked.</li>  
  <li> 
  &quot;Maximum&quot; - displays the maximum value of all the points to be combined</li> 
  <li> 
  &quot;Mean&quot; - displays the mean</li> 
  <li> 
  &quot;Minimum&quot; - displays the minimum of all the points to be combined</li> 
</ul> 
<p> 
<strong>Smoothing window</strong>: When set to a numerical value, this option determines the size,
in pixels, of a smoothing window to be passed over the plot to smooth the edges of the bars or 
lines. This is equivalent to a trend line calculation on the graph. The default setting is 
&quot;OFF&quot;.</p>  
<p> 
<strong>Negate values</strong>: When checked, all values in the wiggle are negated, meaning that 
positive values become negative and vice-versa. This is useful for wiggles representing 
transcription or other activities on the Crick strand. Be aware that wiggles with negative values
are drawn in <strong>altColor</strong>  not <strong>color</strong> as positive values
e.</p> 
<p> 
<strong>Draw indicator lines</strong>:</p> 
<ul> 
  <li> 
  <strong>at y=0.0</strong>: Select the &quot;ON&quot; setting to display a line marking the 0.0 
  position on the graph (default is &quot;OFF&quot;).</li> 
  <li> 
  <strong>at y= </strong>: Select the  &quot;ON&quot; setting to display a line on the graph at 
  the specified numerical value (defaults are &quot;0&quot; and &quot;OFF&quot;). This line can be
   used to mark a significant threshold on the graph.</li> 
</ul> 
<p> 
When you have finished making your configuration changes, click the <em>Apply</em> button to preview
your changes, or click the <em>Submit</em> button to return
to the annotation track display page.</p> 

<a name="TRACK_CONT"></a>
<h3>Annotation track display modes</h3>
<p>
Each annotation track within the window may have up to five display modes:
<p>
<ul class="gbsNoBullet">
  <li>
  <strong>Hide:</strong> the track is not displayed at all. To hide all the annotation tracks, click
  the <em>hide all</em> button. This mode is useful for restricting the display to only those tracks
  in which you are interested. For example, someone who is not interested in SNPs or mouse synteny
  may want to hide these tracks to reduce track clutter and improve speed. There are a few
  annotation tracks that pertain only to one specific chromosome, e.g. Sanger22, Rosetta. In these
  cases, the track and its associated controller will be hidden automatically when the track window
  is not open to the relevant chromosome.</li>
  <li>
  <strong>Dense:</strong> the track is displayed with all features collapsed into a single line.
  This mode is useful for reducing the amount of space used by a track when you don't need
  individual line item details or when you just want to get an overall view of an annotation. For
  example, by opening an entire chromosome and setting the RefSeq Genes track to dense, you can get
  a feel for the known gene density of the chromosome without displaying excessive detail. <br>
  <img src="../../images/wiggle_dense-1.png" width="800px">
  </li>
  <li>
  <strong>Full:</strong> the track is displayed with each annotation feature on a separate line. It
  is recommended that you use this option sparingly, due to the large number of individual track
  items that may potentially align at the selected position. For example, hundreds of ESTs might
  align with a specified gene. When the number of lines within a requested track location exceeds
  250, the track automatically defaults to a more tightly-packed display mode. In this situation,
  you can restore the track display to full mode by narrowing the chromosomal range displayed or by
  using a track filter to reduce the number of items displayed. On tracks that contain only hide,
  dense, and full modes, you can toggle between full and dense display modes by clicking on the
  track's center label.<br>
  <img src="../../images/wiggle_full-1.png" width="800px">
  </li>
  <li>
  <strong>Squish:</strong> the track is displayed with each annotation feature shown separately, but
  at 50% the height of full mode. Features are unlabeled, and more than one may be drawn on the same
  line. This mode is useful for reducing the amount of space used by a track when you want to view a
  large number of individual features and get an overall view of an annotation. It is particularly
  good for displaying tracks in which a large number of features align to a particular section of a
  chromosome, e.g. EST tracks.<br>
  <img src="../../images/wiggle_squish-1.png" width="800px">
  </li>
  <li>
  <strong>Pack:</strong> the track is displayed with each annotation feature shown separately and
  labeled, but not necessarily displayed on a separate line. This mode is useful for reducing the
  amount of space used by a track when you want to view the large number of individual features
  allowed by squish mode, but need the labeling and display size provided by full mode. When the
  number of lines within the requested track location exceeds 250, the track automatically defaults
  to squish display mode. In this situation, you can restore the track display to pack mode by
  narrowing the chromosomal range displayed or by using a track filter to reduce the number of items
  displayed. To toggle between pack and full display modes, click on the track's center label.<br>

  <img src="../../images/wiggle_pack-1.png" width="800px" >
  </li>



<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
