<!DOCTYPE html>
<!--#set var="TITLE" value="Adding Track Metadata" -->
<!--#set var="ROOT" value="../.." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>Adding metadata to tracks</h1>

<h2>Contents</h2>

<h6><a href="#intro">Adding metadata to tracks</a></h6>
<h6><a href="#tagstorm">Tagstorm metadata</a></h6>
<h6><a href="#tabsep">Tab Sep metadata</a></h6>
<h6><a href="#metadata">Previous metadata versions</a></h6>

<a name="intro"></a>
<h2>Adding metadata to tracks</h2>
<p>
Adding metadata to your tracks about cell lines, experimental protocols, or assays can be
accomplished in a number of ways, via the newly supported <em>metadb</em> or <em>metatab</em>
trackDb fields, or via the older style <em>metadata</em> trackDb field. The <em>metadb</em> and
<em>metatab</em> fields link external tagStorm or tabSep metadata files to the data in the hub. The
new formats are preferred over the older <em>metadata</em> field, although the <em>metadata</em>
lines will continue to be supported for track hubs, but no new features will be added as they will
for tagStorm and tabSep files.</p>

<a name="metadata"></a>
<h2>Previous metadata versions</h2>
<p>
Currently, in order to add metadata to your tracks, you must specify all of the metadata key-value
pairs in each stanza of a track that includes metadata, like the last line of the following example:<br>
<pre>
    track oneA-1
    bigDataUrl http://genome-test.soe.ucsc.edu/~chmalee/tagStormHubs/test2/bbi/one-a1.bb
    parent oneA off
    type bigBed 3
    shortLabel oneA-1
    longLabel oneA-1
    color 250,108,0
    visilibility dense
    metadata  differentiation=10hour treatment=X donor=A lab=UCSC data_set_id=ucscTest1 access=group assay=long-RNA-seq enriched_in=exon life_stage=postpartum species="Homo sapiens" ucsc_db=hg38
</pre>
Each track must have a separate <em>metadata</em> field and its own list of key-values, which can become cumbersome when each track in a group all share a common subset of metadata. For instance, if there are 10 tracks in a composite or multiWig, where each subtrack only differs in the &quot;differentiation&quot; tag, it would be more convenient to have a shared set of metadata and then specify the differences for each track. This is the motivation behind the tagStorm format, described below.
</p>

<a name="tagstorm"></a>
<h2>Tagstorm metadata</h2>
<p>
The tagStorm format is a plaintext file similar to the <em>trackDb.txt</em> file that describes all of
the tracks in a <a href="hgTrackHubHelp.html">track hub</a>, in that both are files where the
first word in a  line is the tag and the rest of the line is the value, and different stanza's are
line delimited. TagStorm's are also similar to a spreadsheet, where a tag corresponds to a column
and a stanza to an entire row. Here is a canonical tagStorm example:<br>
<pre>
lab UCSC
data_set_id ucscTest1
access group
assay long-RNA-seq
enriched_in exon
life_stage postpartum
species Homo sapiens
ucsc_db hg38

    treatment X
    donor A

        differentiation 10 hour
        meta ucsc1_1

        differentiation 1 day
        meta ucsc1_4

        differentiation 5 days
        meta ucsc1_7

    treatment Y

        donor B

            differentiation 10 hour
            meta ucsc1_2

            differentiation 1 day
            meta ucsc1_5

            differentiation 5 days
            meta ucsc1_8

        donor C

            differentiation 10 hour
            meta ucsc1_3

            differentiation 1 day
            meta ucsc1_6

            differentiation 5 days
            meta ucsc1_9
</pre>
<p>
Each stanza, such as &quot;donor B&quot;, inherits from any stanzas above it at the right indentation level, and is a parent to
stanzas beneath.</p>

<a name="tabsep"></a>
<h2>Tab sep metadata</h2>
<p>
Column or tab separated metadata can be useful to store computer readable information as an array. 
</p>
<pre>
#lab	data_set_id	access	assay	enriched_in	life_stage	species	ucsc_db	treatment	donor	differentiation	meta
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38		
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38	X	A
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38	X	10 hour	ucsc1_1
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38	X	1 day	ucsc1_4
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38	X	5 days	ucsc1_7
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38	Y	
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38	Y	B
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38	Y	10 hour	ucsc1_2
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38	Y	1 day	ucsc1_5
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38	Y	5 days	ucsc1_8
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38	Y	C
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38	Y	10 hour	ucsc1_3
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38	Y	1 day	ucsc1_6
UCSC	ucscTest1	group	long-RNA-seq	exon	postpartum	Homo sapiens	hg38	Y	5 days	ucsc1_9
</pre>

<a name="License"></a>
<h2>Licensing information</h2>
<p>
GBiB is free for non-profit academic research and for personal use. Corporate use requires a
license, setup fee and annual payment. To purchase a license or download the GBiB, visit the
<a href="http://genome-store.ucsc.edu/" target="_blank">Genome Browser store</a>.</p>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
