<h2>Description</h2>

<p> This track shows potential contact residues for ligands, inferred from
structures in the PDB database by Alastair Fyfe, UCSC.
</p>

<h2>Display Conventions and Configuration</h2>

<p>
Genomic locations of contact residues are highlight with thick blocks, 
that look identical to exons. Contact residues of the same PDB structure
are connected by thin intron lines.
</p>

<h2>Methods</h2>
<p>
PDB SEQRET protein sequences were aligned to the genome with tblastn.
Ligands were determined manually, all amino
acids closer than 10 Angstroms were obtained with the software XXX.
The positions of the close amino acids on the genome were determined 
with the tblastn alignment. The software is available at XXXX.
</p>

<h2>Data Access</h2>
<p>
The raw data can be explored interactively with the
<a href="hgTables">Table Browser</a> or combined with other datasets in the
<a href="hgIntegrator">Data Integrator</a> tool.
For automated analysis, the genome annotation is stored in
a bigBed file that can be downloaded from
<a href="http://hgdownload.soe.ucsc.edu/gbdb/wuhCor1/bbi/contacts.bb">the download server</a>.</p>
<p>
Annotations can
be converted from binary to ASCII text by our command-line tool bigBedToBed.
Instructions for downloading this command can be found on our
<a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">utilities page</a>.
The tool can also be used to obtain features within a given range without downloading the file,
for example:</p>
<code>bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/wuhCor1/bbi/contacts.bb -chrom=NC_045512v2 -start=0 -end=29902 stdout</code>
<p>
Please refer to our
<a href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome">mailing list archives</a>
for questions, or our
<a href="../../FAQ/FAQdownloads.html#download36">Data Access FAQ</a>
for more information.</p>

<h2>References</h2>

XXX?
