<h2>Description</h2>
<p>
This composite track indicates the immune epitope predictions
for <em>B cells</em>, <em>CD4 T-cells</em> and <em>CD8 T-cells</em>, using these software packages:
B cells = BebiPred 2.0, CD4 = IEDB Tepitool, CD8 = NetMHCpan4.0EL</p>
<p>
From the publication:
<a href='https://www.biorxiv.org/content/10.1101/2020.02.12.946087v1'
target=_blank>
Candidate targets for immune responses to 2019-Novel Coronavirus (nCoV):
sequence homology- and bioinformatic-based predictions</a>,
<em>Alba Grifoni, John Sidney, Yun Zhang, Richard H Scheuermann, Bjoern Peters,
Alessandro Sette</em> (doi: https://doi.org/10.1101/2020.02.12.946087)
</p>
<p>The prediction of epitopes for CD8 T-cells was run for the following HLA alleles, as they have a worldwide population
frequency &gt; 6%<p> <table class='bedExtraTbl'><tr><td>HLA
alleles</td><td>Frequency in worldwide
population</td></tr><tr><td>HLA-A*01:01</td><td>16.2</td></tr><tr><td>HLA-A*02:01</td><td>25.2</td></tr><tr><td>HLA-A*03:01</td><td>15.4</td></tr><tr><td>HLA-A*11:01</td><td>12.9</td></tr><tr><td>HLA-A*23:01</td><td>6.4</td></tr><tr><td>HLA-A*24:02</td><td>16.8</td></tr><tr><td>HLA-B*07:02</td><td>13.3</td></tr><tr><td>HLA-B*08:01</td><td>11.5</td></tr><tr><td>HLA-B*35:01</td><td>6.5</td></tr><tr><td>HLA-B*40:01</td><td>10.3</td></tr><tr><td>HLA-B*44:02</td><td>9.2</td></tr><tr><td>HLA-B*44:03</td><td>7.6</td></tr></table>
</p>
<h2>Summary</h2>
<p>
We identified potential targets for immune responses to 2019-nCoV
and provide essential information for understanding human immune responses
to this virus and evaluation of diagnostic and vaccine candidates.
</p>
<h2>Abstract</h2>
<p>
Effective countermeasures against the recent emergence and rapid expansion
of the 2019-Novel Coronavirus (2019-nCoV) require the development of data
and tools to understand and monitor viral spread and immune responses.
However, little information about the targets of immune responses to
2019-nCoV is available. We used the Immune Epitope Database and Analysis
Resource (IEDB) resource to catalog available data related to other
coronaviruses, including SARS-CoV, which has high sequence similarity to
2019-nCoV, and is the best-characterized coronavirus in terms of epitope
responses. We identified multiple specific regions in 2019-nCoV that have
high homology to SARS virus. Parallel bionformatic predictions identified a
priori potential B and T cell epitopes for 2019-nCoV. The independent
identification of the same regions using two approaches reflects the high
probability that these regions are targets for immune recognition of 2019-nCoV.
</p>

<h2>Credits</h2>
Data collected by <em>Arkal Arjun Rao</em> for the
<a href='https://niksgourakis.chemistry.ucsc.edu/'
target=_blank>Sgourakis Research Group</a>, U.C. Santa Cruz

<h2>References</h2>
<p>
Grifoni et al.: <a href="https://www.biorxiv.org/content/10.1101/2020.02.12.946087v1" target=_blank>Candidate targets for immune responses to 2019-Novel Coronavirus (nCoV):sequence homology- and bioinformatic-based predictions, Biorxiv 2020</a>
</p>

<h2>Funding</h2>
<p>
The SARS-CoV-2 genome browser and data annotation tracks are funded by
generous individual donors including Pat & Rowland Rebele.</p>
