<h2>Description</h2>
<p>
<a href="https://nextstrain.org/ncov" target=_blank>Nextstrain.org</a> displays
data about protein-coding variants that occur in the current 2019/2020 outbreak. 
This track shows all the Nextstrain sequence variants. It is updated daily.
The original source of the full viral sequences are 
<a href="https://www.ncbi.nlm.nih.gov/genbank/" target=_blank>NCBI Genbank</a> and 
<a href="https://gisaid.org" target=_blank>GISAID</a>.
</p>

<p>
Use this track in conjunction with the tracks Nextstrain Genes</em> (parts of
genome covered by Nextstrain) and <em>Nextstrain Clades</em> (the main classes
of variants as defined by Nextstrain).
</p>

<h2>Summary</h2>
<p>
</p>

<h2>Methods</h2>
<p>Data was downloaded from Nextstrain.org, from the file ncov.json and converted to VCF.</p>

<h2>Data Access</h2>
<p>You can download the VCF file underlying this track from our <a href="https://hgdownload.soe.ucsc.edu/gbdb/wuhCor1/bbi/" target=_blank>Download Server.</a></p>
<p>
<p>
The raw data can be explored interactively with the <a href="../cgi-bin/hgTables">Table Browser</a>
or the <a href="../cgi-bin/hgIntegrator">Data Integrator</a>. The data can be
accessed from scripts through our <a href="https://api.genome.ucsc.edu">API</a>, the track name is
"nextstrainVars".
<p>

<h2>Credits</h2>
This work is made possible by the open sharing of genetic data by research
groups from all over the world. We gratefully acknowledge their contributions.

<h2>Data usage policy</h2>
The data presented here is intended to rapidly disseminate analysis of
important pathogens. Unpublished data is included with permission of the data
generators, and does not impact their right to publish. Please contact the
respective authors (available via the <a href="https://nextstrain.org/c163de10-289e-407f-a005-449f5725c2dd">Nextstrain
TSV file</a>) if you intend to carry out further research using their data.
Derived data, such as phylogenies, can
be downloaded below - please contact the relevant authors where appropriate.

<h2>References</h2>
<p>
Hadfield J, Megill C, Bell SM, Huddleston J, Potter B, Callender C, Sagulenko P, Bedford T, Neher
RA.
<a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/bty407"
    target="_blank">
    Nextstrain: real-time tracking of pathogen evolution</a>.
<em>Bioinformatics</em>. 2018 Dec 1;34(23):4121-4123.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29790939" target="_blank">29790939</a>; PMC: <a
    href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6247931/" target="_blank">PMC6247931</a>
</p>
<p>
Sagulenko P, Puller V, Neher RA.
<a href="https://academic.oup.com/ve/article-lookup/doi/10.1093/ve/vex042" target="_blank">
TreeTime: Maximum-likelihood phylodynamic analysis</a>.
<em>Virus Evol</em>. 2018 Jan;4(1):vex042.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29340210" target="_blank">29340210</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5758920/" target="_blank">PMC5758920</a>
</p>

