<h2>Description</h2>
<p>
<a href="https://nextstrain.org/ncov" target=_blank>Nextstrain.org</a> displays
data about variants that occur in the current 2019/2020 outbreak.
Nextstrain has a powerful user interface for viewing the evolutionary tree
that it infers from the patterns of variants in sequences worldwide, but
does not offer a detailed plot of variants along the genome, so we have processed
their data into this track to display the variants of each sample that Nextstrain
has obtained from
<a href="https://gisaid.org" target=_blank>GISAID</a>.
</p>
<p>
Click on a variant to see more details including protein change (if applicable;
protein changes use gene names in the <em>Nextstrain Genes</em> track) and the
genotype for each GISAID sample.
</p>

<h2>Display Conventions</h2>
<p>
In &quot;dense&quot; mode, a vertical line is drawn at the position of each variant.
In &quot;pack&quot; mode, the display shows a plot of all samples' variants, with samples
ordered using Nextstrain's phylogenetic tree in order to highlight patterns of linkage.</p>
<p>
Each sample is placed in a horizontal row of pixels; when the number of
samples exceeds the number of vertical pixels for the track, multiple
samples fall in the same pixel row and pixels are averaged across samples.</p>
<p>
Each variant is a vertical bar with white (invisible) representing the reference allele
and black representing the non-reference allele(s).
Tick marks are drawn at the top and bottom of each variant's vertical bar
to make the bar more visible when most alleles are reference alleles.
</p>

<h2>Methods</h2>
<p>Nextstrain downloads SARS-CoV-2 genomes from
  <a href="https://gisaid.org/" target=_blank>GISAID</a>
  as they are submitted by labs worldwide.
  The sequences are processed by an
  <a href="https://github.com/nextstrain/ncov" target=_blank>automated pipeline</a>
  and annotations are written to the data file
  <a href="http://data.nextstrain.org/ncov.json" target=_blank>ncov.json</a>.
UCSC downloads ncov.json and extracts annotations for display.</p>

<h2>Data Access</h2>
<p>
You can download the bigBed file underlying this track (nextstrainSamples*) from our
<a href="https://hgdownload.soe.ucsc.edu/gbdb/wuhCor1/nextstrain/" 
target=_blank>Download Server</a>. The data can be explored interactively with the 
<a href="../cgi-bin/hgTables">Table Browser</a>
or the <a href="../cgi-bin/hgIntegrator">Data Integrator</a>. The data can be
accessed from scripts through our <a href="https://api.genome.ucsc.edu">API</a>.
The original source file can be accessed
<a href="http://data.nextstrain.org/ncov.json" 
target=_blank>directly from NextStrain</a> in JSON format.</p>

<h2>Credits</h2>
<p>This work is made possible by the open sharing of genetic data by research
groups from all over the world. We gratefully acknowledge their contributions.
Special thanks to
  <a href="https://nextstrain.org/ncov" target=_blank>nextstrain.org</a> for making its
  <a href="http://data.nextstrain.org/ncov.json" target=_blank>data file</a> freely available.
</p>

<h2>Data usage policy</h2>
The data presented here is intended to rapidly disseminate analysis of
important pathogens. Unpublished data is included with permission of the data
generators, and does not impact their right to publish. Please contact the
respective authors (available via the
<a href="https://github.com/nextstrain/ncov/blob/master/data/metadata.tsv"
   target=_blank>Nextstrain metadata.tsv file</a>)
if you intend to carry out further research using their data.
Derived data, such as phylogenies, can be downloaded from
<a href="https://nextstrain.org/ncov" target=_blank>nextstrain.org</a>
(see "DOWNLOAD DATA" link at bottom of page) -
please contact the relevant authors where appropriate.

<h2>References</h2>
<p>
Hadfield J, Megill C, Bell SM, Huddleston J, Potter B, Callender C, Sagulenko P, Bedford T, Neher
RA.
<a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/bty407"
    target="_blank">
    Nextstrain: real-time tracking of pathogen evolution</a>.
<em>Bioinformatics</em>. 2018 Dec 1;34(23):4121-4123.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29790939" target="_blank">29790939</a>; PMC: <a
    href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6247931/" target="_blank">PMC6247931</a>
</p>
<p>
Sagulenko P, Puller V, Neher RA.
<a href="https://academic.oup.com/ve/article-lookup/doi/10.1093/ve/vex042" target="_blank">
TreeTime: Maximum-likelihood phylodynamic analysis</a>.
<em>Virus Evol</em>. 2018 Jan;4(1):vex042.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29340210" target="_blank">29340210</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5758920/" target="_blank">PMC5758920</a>
</p>
