<h2>Description</h2>

<p>
This track shows where IgG and IgM antibodies bind the SARS-CoV-2 proteins, on
data from 10 Covid patients and 12 healthy controls, determined with a SARS-CoV-2 protein peptide microarray.</p>

<h2>Display Conventions and Configuration</h2>

<p>
Genomic locations of peptides that were spotted on the array are highlighted. Because these peptides overlap, the tracks default to dense mode and the sequence is shown as labels drawn onto the rectangles, but only visible on high zoom levels. Put any track into pack mode to fully see the sequence.</p>

<p>The color is assigned based on the Z-Score, without any other normalization. Blue with decreasing intensity is assigned to the values -5 to 0, white is 0, and red colors with increasing intensity are used for the values 0-3.5, very similar as in the original publication figures.

<h2>Methods</h2>
<p>
Supplemental files were converted from Excel, rearranged and run through the command line script bigHeat to create a heatmap-like display, with multiplication factors of 2 for negative values, 0.285 for positive values. For better visibility, colormap seismic from matplotlib was used from 0.1 to 0.9 and the range of values after multiplication were restricted to the limits -1 to 1 to address outliers. Like all tracks, the exact commands are documented in our <a href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg/makeDb/doc/wuhCor1/pbm.txt" target=_blank>makeDoc text files</a>.
</p>

<h2>Data Access</h2>
<p>
The raw data can be explored interactively with the
<a href="hgTables">Table Browser</a> or combined with other datasets in the
<a href="hgIntegrator">Data Integrator</a> tool.
For automated analysis, the genome annotation is stored in
a bigBed file that can be downloaded from
<a href="http://hgdownload.soe.ucsc.edu/gbdb/wuhCor1/bbi/artic.bb">the download server</a>.</p>
<p>
Annotations can
be converted from binary to ASCII text by our command-line tool bigBedToBed.
Instructions for downloading this command can be found on our
<a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">utilities page</a>.
The tool can also be used to obtain features within a given range without downloading the file,
for example:</p>
<code>bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/wuhCor1/bbi/artic.bb -chrom=NC_045512v2 -start=0 -end=29902 stdout</code>
<p>
Please refer to our
<a href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome">mailing list archives</a>
for questions, or our
<a href="../../FAQ/FAQdownloads.html#download36">Data Access FAQ</a>
for more information.</p>

<h2>References</h2>
<p>
Hongye Wang, Xian Wu, Xiaomei Zhang, Xin Hou, Te Liang, Dan Wang, Fei Teng, Jiayu Dai, Hu Duan, Shubin Guo, Yongzhe Li, and Xiaobo Yu
<a href="https://pubs.acs.org/doi/pdf/10.1021/acscentsci.0c00742" target="_blank">SARS-CoVâ€‘2 Proteome Microarray for Mapping COVID-19 Antibody Interactions at Amino Acid Resolution</a>.

<em>ACS Central Science</em>. 2020
DOI: <a href="http://dx.doi.org/10.1021/acscentsci.0c00742" target="_blank">10.1021/acscentsci.0c00742</a>
</p>

