<h2>Description</h2>
<p class="p4">These tracks show curated SARS-CoV-2 protein-coding genes conserved within the Sarbecovirus subgenus as determined using PhyloCSF [1], FRESCo [2], and other comparative genomics methods, consistent with experimental evidence in SARS-CoV-2. For a complete description of the evidence, see [3].</p>
<ul class="ul1">
  <li class="li4">
      The <b>PhyloCSF Genes</b> track shows the conserved protein-coding genes, namely ORF1a, ORF1ab, S, ORF3a, ORF3c (a.k.a. ORF3h, ORF3a*, and 3a.iORF1), E, M, ORF6, ORF7a, ORF7b, ORF8, N, and ORF9b (a.k.a. ORF9a).
  </li>
  <li class="li4">
      The <b>PhyloCSF Rejected Genes</b> track shows other genes that have been proposed that do not show the signature of conserved protein-coding genes or persuasive experimental evidence, and are thus unlikely to be actual protein-coding genes, namely ORF3b-2, ORF3b, ORF14 (a.k.a. ORF9b), and ORF10. (Two different ORFs have been referred to as ORF3b. We refer to the one with coordinates 25524-25697 as ORF3b-2. The other, with coordinates 25814-26281, is the ortholog of SARS-CoV-1 ORF3b, but is not an ORF in SARS-CoV-2 due to in-frame stop codons.)
  </li>
</ul>

<h2>Methods</h2>
<p class="p4">See [3].</p>

<h2>Credits</h2>
<p class="p4">Questions should be directed to <a href="mailto:iljungr@csail.mit.edu"><span class="s1">Irwin Jungreis</span></a>.</p>
<p class="p4">If you use the SARS-CoV-2 PhyloCSF Genes Track Hub, please cite Jungreis et al. 2020 [3].</p>
<h2>Reference</h2>

<p class="p5"><span class="s3">[1] </span>Lin MF, Jungreis I, and Kellis M (2011). PhyloCSF: a comparative genomics method to distinguish protein-coding and non-coding regions. <a href="http://bioinformatics.oxfordjournals.org/content/27/13/i275.abstract"><span class="s4"><i>Bioinformatics</i></span><span class="s1"> 27:i275-i282</span></a> (ISMB/ECCB 2011).</p>

<p class="p5"><span class="s3">[2] </span>Sealfon RS, Lin MF, Jungreis I, Wolf MY, Kellis M, Sabeti PC (2015). FRESCo: finding regions of excess synonymous constraint in diverse viruses. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4376164/"><span class="s4"><i>Genome Biol. doi: 10.1186/s13059-015-0603-7</i></span></a>.</p>

<p class="p5"><span class="s3">[3] </span>Jungreis I, Sealfon R, Kellis M (2020). Sarbecovirus comparative genomics elucidates gene content of SARS-CoV-2 and functional impact of COVID-19 pandemic mutations. <a href="https://www.biorxiv.org/content/10.1101/2020.06.02.130955v1"><span class="s4"><i>bioRxiv doi:10.1101/2020.06.02.130955</i></span></a>.</p>
