<h2>Description</h2>
This track contains deep mutational scanning data measuring the effect of the change in expression from a wild type allele to a mutant allele.
The authors use a yeast display system to experimentally measure the effect of all possible point (amino acid) RBD mutations on protein expression &
ACE2 affinity.

<h2>Display Conventions and Configuration</h2>
<p>Each subtrack contains all the scores representing mutations to a particular amino acid (each annotation is a S codon).  For instance the A subtrack measures the change in expression of
S RBD expression if the annotated amino acid is mutated to alanine (if the wildtype amino acid is A, then the score is 0). A positive score indicates increased expression a negative score is a loss of expression.

<p>Please see the interactive heatmap generated by the authors at <a href="https://jbloomlab.github.io/SARS-CoV-2-RBD_DMS/"> this link</a>
<p>Structural visualizations of the data are available from the authors via dms-view <a href = "https://jbloomlab.github.io/SARS-CoV-2-RBD_DMS/structures"> here.</a>


<h2>Methods</h2>
Table S2 from Starr et al, was downloaded and parsed into bedGraph format using the average value of both replicates reported.  All NA values were filtered out.


<h2>References</h2>
<p>
Starr TN, Greaney AJ, Hilton SK, Ellis D, Crawford KHD, Dingens AS, Navarro MJ, Bowen JE, Tortorici
MA, Walls AC <em>et al</em>.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/32841599" target="_blank">
Deep Mutational Scanning of SARS-CoV-2 Receptor Binding Domain Reveals Constraints on Folding and
ACE2 Binding</a>.
<em>Cell</em>. 2020 Sep 3;182(5):1295-1310.e20.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32841599" target="_blank">32841599</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7418704/" target="_blank">PMC7418704</a>
</p>

