<!DOCTYPE html>
<!--#set var="TITLE" value="CIRM Stem Cell Hub at UCSC Homepage" -->

<!--#include virtual="setRoot.html" -->

<!--#include virtual="${ROOT}inc/header.html" -->

<div class="cirm-page-body">
  <div class="cirm-page-title back-light-green">
    <h5 class="cirm-page-title-text">Tools from the CESCG groups</h5>
  </div>
  <div class="row no-gutters">
    <div class="card-deck">
      <a name="CellBrowser"></a>
      <div class="card">
        <div class="card-body text-center">
          <img class="card-top-img" style="max-width: 300px"
          src="https://cells.ucsc.edu/cortex-dev/thumb.png">
          <p class="lead text-center"><span style="color: #1E90FF">Cell Browser:</span> Visualize single-cell clustering data</p>
          <ul class="list-group list-group-flush text-left">
            <li class="list-group-item">The Cell Browser allows for the visualization of single-cell
            clustering data with the option to color cell by metadata attributes such as sex, age,
            and gene expression. A number of published datasets are available through the Cell
            Browser website or set up a Cell Browser for your own data using the code on GitHub.
            </li>
            <li class="list-group-item"><a href="https://cells.ucsc.edu" target="_blank">Cell
            Browser Website</a> <i class="fas fa-external-link-alt" aria-hidden="true"></i></li>
            <li class="list-group-item"><a href="https://github.com/maximilianh/cellBrowser"
            target="_blank">Cell Browser GitHub</a> <i class="fab fa-github" aria-hidden="true"></i>
            </li>
          </ul>
        </div>
        <div class="card-footer">
          <small class="text-muted">
          The Cell Browser was created by Max Haeussler.
          </small>
        </div>
      </div>
      <a name="PluriTest"></a>
      <div class="card">
        <div class="card-body text-center">
          <img class="card-top-img" style="max-width: 300px"
          src="<!--#echo var='ROOT'-->images/pluritestWelcome.png">
          <p class="lead"><span style="color: #1E90FF">PluriTest:</span> Check for Pluripotency</p>
          <ul class="list-group list-group-flush text-left">
            <li class="list-group-item">
            Pluritest allows you check your samples for pluripotentcy through uploading RNA-seq
            fastq files or Affymetrix and Illumina arrays.
            Note: Use of Pluritest requires you to register an account.
            </li>
            <li class="list-group-item">
            M&#252;ller FJ, Schuldt BM, Williams R, Mason D, Altun G, Papapetrou EP, Danner S, Goldmann JE,
            Herbst A, Schmidt NO <em>et al</em>.
            <a href="http://dx.doi.org/10.1038/nmeth.1580" target="_blank">
            A bioinformatic assay for pluripotency in human cells</a>.
            <em>Nat Methods</em>. 2011 Apr;8(4):315-7.
            PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21378979" target="_blank">21378979</a>; PMC: <a
            href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3265323/" target="_blank">PMC3265323</a>
            </li>
            <li class="list-group-item"><a href="https://www.pluritest.org/" target="_blank">PluriTest
            Website</a> <i class="fas fa-external-link-alt" aria-hidden="true"></i></p>
            </li>
          </ul>
        </div>
        <div class="card-footer">
          <small class="text-muted">
          PluriTest was created by Roy Williams and the <a
          href="https://www.scripps.edu/loring/" target="_blank">Jeanne Loring Lab</a>.
          </small>
        </div>
      </div>
    </div>
  </div>
  <br>
  <div class="row no-gutters">
    <div class="card-deck">
      <a name="Pgltools"></a>
      <div class="card">
        <div class="card-body text-center">
          <span class="padded-top-img">
            <img class="img-responsive" style="max-width: 300px"
            src="https://raw.githubusercontent.com/billgreenwald/pgltools/master/Images/logo.png">
          </span>
          <p class="lead"><span style="color: #1E90FF">Pgltools:</span> Manipulate Hi-C and ChIA-PET
          data</p>
          <ul class="list-group list-group-flush text-left">
            <li class="list-group-item">
            The pgltools package gives you a number of utilities for manipulating files in the
            <a href="https://github.com/billgreenwald/pgltools#the-pgl-file-format"
            target="_blank">&quot;paired genomic loci&quot;</a> format, common in both Hi-C and 
            ChIA-PET experiments. These utilities allow you to things such as convert pgl files
            to a number of other formats, compare and contrast multiple pgl files, and other basic
            operations like sorting or merging files.
            </li>
            <li class="list-group-item">
            Greenwald WW, Li H, Smith EN, Benaglio P, Nariai N, Frazer KA.
            <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1621-0"
            target="_blank">
            Pgltools: a genomic arithmetic tool suite for manipulation of Hi-C peak and other chromatin
            interaction data</a>.
            <em>BMC Bioinformatics</em>. 2017 Apr 7;18(1):207.
            PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28388874" target="_blank">28388874</a>; PMC: <a
            href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5384132/" target="_blank">PMC5384132</a>
            </li>
            <li class="list-group-item"><a href="https://github.com/billgreenwald/pgltools"
            target="_blank">Pgltools GitHub</a> <i class="fab fa-github" aria-hidden="true"></i></li>
          </ul>
        </div>
        <div class="card-footer">
          <small class="text-muted">
          The Pgltools package was created by Bill Greenwald.
          </small>
        </div>
      </div>
      <a name="SCIMITAR"></a>
      <div class="card">
        <div class="card-body text-center">
          <span class="padded-top-img">
            <img src="">
          </span>
          <p class="lead"><span style="color: #1E90FF">SCIMITAR:</span> Single Cell Inference of
          MorphIng Trajectories and their Associated Regulation</p>
          <ul class="list-group list-group-flush text-left">
            <li class="list-group-item">The SCIMITAR package contains a number of tools to help
            scientists analyze cell trajectory maps based on single cell sequencing data.</li>
            <li class="list-group-item">
            Cordero P, Stuart JM.
            <a href="http://dx.doi.org/10.1142/9789813207813_0053" target="_blank">
            Tracing co-regulatory network dynamics in noisy, single-cell transcriptome trajectories</a>
            <i class="far fa-file-pdf" aria-hidden="true"></i>.
            <em>Pac Symp Biocomput</em>. 2016;22:576-587.
            PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27897008" target="_blank">27897008</a>; PMC: <a
            href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5203771/" target="_blank">PMC5203771</a>
            </li>
            <li class="list-group-item">
            <a href="https://github.com/dimenwarper/scimitar" target="_blank">SCIMITAR GitHub</a>
            <i class="fab fa-github" aria-hidden="true"></i></p>
            </li>
          </ul>
        </div>
        <div class="card-footer">
          <small class="text-muted">
          SCIMITAR was created Pablo Cordero.
          </small>
        </div>
      </div>
    </div>
  </div>
  <br>
  <div class="row no-gutters">
    <div class="card-deck">
      <a name="tagStorm"></a>
      <div class="card">
        <div class="card-body text-center">
          <img class="card-top-img" style="max-width: 300px"
          src="<!--#echo var='ROOT'-->images/tagStormImage.png">
          <p class="lead"><span style="color: #1E90FF">Tag Storm:</span> Metadata made
          simple</p>
          <ul class="list-group list-group-flush text-left">
            <li class="list-group-item">
            <p>
            The tag storm format offers an easy way to describe
            a hierarchical set of metadata for your data. The tag storm format is made up of
            a number of key/value pairs spread across stanzas at various indentation levels.
            Lower indentation levels inherit information from higher levels, so the most general
            metadata, e.g. lab or dataset names, are generally found at the highest level and
            more specifc metadata, e.g. sex or file name, are found at the lower level.</p>
            <p>
            The Tag Storm GitHub page contains more information about the format as well as a
            suite of tools to work with and manipulate tag storm files.
            </li>
            <li class="list-group-item">
            <a href="https://github.com/ucscGenomeBrowser/kent/tree/master/src/tagStorm"
            target="_blank">Tag Storm GitHub</a> <i class="fab fa-github" aria-hidden="true"></i></p>
            </li>
          </ul>
        </div>
        <div class="card-footer">
          <small class="text-muted">
          TagStorm format and utilities were developed by Jim Kent.
          </small>
        </div>
      </div>
      <a name="RIGGLE"></a>
      <div class="card">
        <div class="card-body text-center">
          <img class="card-top-img" style="max-width: 300px"
          src="<!--#echo var='ROOT'-->images/RIGGLE_demo.png">
          <p class="lead"><span style="color: #1E90FF">RIGGLE:</span> Regulator Inference by
          Graph-Guided LASSO Estimation</p>
          <ul class="list-group list-group-flush text-left">
            <li class="list-group-item">
            <p>
            RIGGLE (Regulatory Inference by Graph-Guided LASSO Estimation)
            is a machine learning framework designed to discover the transcription factor regulators
            of a cell development trajectory by taking in expression data, known transcription
            factor targets, and a cell development graph.</p>
            <p>
            This process produces predictions for the transcription factor's
            activities in each of the cell types represented in the graph based on the
            coordinated expression of its targets, while respecting the developmental
            relationships between the cells.</p>
            </li>
          </ul>
        </div>
        <div class="card-footer">
          <small class="text-muted">
          RIGGLE was created by Dan Carlin.
          </small>
        </div>
      </div>
    </div>
  </div>
  <br>
  <div class="row no-gutters">
    <div class="card-deck">
      <a name="samplePsychic"></a>
      <div class="card">
        <div class="card-body text-center">
          <span class="card-top-img">
            <div id="carouselInd" class="carousel slide" data-ride="carousel" style="/*background-color: black*/">
              <div class="carousel-inner">
                <div class="carousel-item active">
                  <img style="max-height: 200px"
                  src="<!--#echo var='ROOT'-->images/SamplePsychicReportCard.png"
                  alt="Report Card view, showing matches grouped by sample. Has CSV download.">
                </div>
                <div class="carousel-item">
                  <img style="max-height: 200px"
                  src="<!--#echo var='ROOT'-->images/SamplePsychicDetailedView.png"
                  alt="By-sample view. Similar to Report Card but easier to navigate.">
                </div>
                <div class="carousel-item">
                  <img style="max-height: 200px"
                  src="<!--#echo var='ROOT'-->images/SamplePsychicPlot.png"
                  alt="Experimental tSNE clustering of input samples.">
                </div>
              </div>
              <a class="carousel-control-prev" href="#carouselInd" role="button" data-slide="prev">
              <i class="fas fa-angle-left fa-2x text-primary" aria-hidden="true"></i>
              </a>
              <a class="carousel-control-next" href="#carouselInd" role="button" data-slide="next">
              <i class="fas fa-angle-right fa-2x text-primary" aria-hidden="true"></i>
              </a>
            </div>
          </span>
          <p class="lead"><span style="color: #1E90FF">Sample Psychic:</span> See how your
          cells compare</p>
            <ul class="list-group list-group-flush text-left">
            <li class="list-group-item">
            <p>
            Sample Psychic can take your gene expression data and
            compare it to curated references.
            This comparison is used to produce a scorecard which shows how closely your data maps
            up to the curated cell types.</p>
            <p>
            The Sample Psychic web application allows
            users to upload gene expression data and gain insights into those samples by
            applying suites of pre-trained concept classifiers to each sample. The
            resulting set of classification scores is presented as a report card that can
            be used to validate the identity of the samples and explore similarity between
            these samples and target concepts represented in in the Sample Psychic suite.
            For CIRM, classifiers were built for both fetal tissue clusters and hand
            labeled brain regions. One binary classifier is created for tissue
            type/cluster. A total of 96 fetal tissue clusters are represented in the
            Sample Psychic classifier compendium along with 30 classifiers representing
            hand-labeled brain regions. The pattern of individual classifier scores can
            give a picture of how user samples map to brain regions and to previously
            analyzed clusters.</p></li>
            <li class="list-group-item"><a href="https://samplepsychic.ucsc.edu/"
            target="_blank">Sample Psychic Website</a> <i class="fas fa-external-link-alt"
            aria-hidden="true"></i></p></li>
          </ul>
        </div>
        <div class="card-footer">
          <small class="text-muted">
          Sample Psychic was created by James Durbin.
          </small>
        </div>
      </div>
      <a name="MISCE"></a>
      <div class="card">
        <div class="card-body text-center">
          <img class="card-img-top" style="max-width: 300px"
          src="<!--#echo var='ROOT'-->images/misce1.png">
          <p class="lead"><span style="color: #1E90FF">MISCE:</span> A Minimum Information
          About a Stem Cell Experiment</p>
          <ul class="list-group list-group-flush text-left">
            <li class="list-group-item">
              <p>
              MISCE, or Minimum Information about a Stem Cell Expriment, is a
              spreadsheet that attempts to collect a rich set of useful information (metadata) about
              any experiment involving stem cells. The hope is that in using MISCE, the detailed
              metadata will make the experimental data useful to others attempting to use it
              in the future.</p>
              <p>
              MISCE consists of a number of different modules with each module describing a
              different experimental process, e.g. celluar reprogramming or RNA sequencing,
              and all of the important information related to this process, e.g. &quot;Biosample
              disease stage&quot; or &quot;Assay platform&quot;. Each module can
              be included or excluded based on the type of experiment being performed. For
              example, you might not include the &quot;Assay_DNAMethyl&quot; module if you
              are not performing DNA methylation assays as part of your experiment.</p>
            </li>
            <li class="list-group-item">
              <a href="<!--#echo var='ROOT'-->MISCEv1.1.pdf">PDF of MISCE modules</a>
              <i class="far fa-file-pdf" aria-hidden="true"></i>
            </li>
          </ul>
        </div>
        <div class="card-footer">
          <small class="text-muted">
          MISCE was created by Sagar Jain and Richard Scheuermann;
          it is currently maintained by UCSC.
          </small>
        </div>
      </div>
    </div>
  </div>
</div>

<!--#include virtual="${ROOT}inc/footer.html" -->
