<H2>Description</H2>
<P>
The ${longLabel} track set shows various loss of function constraint metrics as predicted by the
<a href="https://gnomad.broadinstitute.org">Genome Aggregation Database (gnomAD)</a> for
gnomAD v2.1.1. The metrics include:
<ul>
  <li>Observed and expected variant counts per transcript/gene
  <li>Observed/Expected  ratio (O/E)
  <li>Z-scores of the observed counts compared to expected
  <li>Probability loss of function intolerance (pLI), for loss-of-function variation only
  <li>Chi-Squared difference of observed to expected counts, for the missense constraint track only
</ul>
The O/E ratio is a continuous measurement of how tolerant a gene or transcript is to a certain
class of variation. Each item in this track shows the O/E ratio for three different types of
variation: missense, synonymous, and loss-of-function. The confidence interval is displayed along
with the O/E ratio to better assist interpretation of the scores. Please see the Methods section
below for more information about how the scores were calculated.
</p>

<h2>Display Conventions and Configuration</h2>
<p>
There are two &quot;groups&quot; of tracks in this set:
<ol>
  <li><b>LoF tracks</b>: Loss of Function Constraint metrics at the whole gene (canonical transcript)
    level or whole transcript level. The Gene LoF track displays metrics for a canonical transcript 
    per gene defined as the longest isoform. Items on both tracks are shaded according to the pLI 
    score, with outlier items shaded in grey.<br>
  <img src="../images/pLIscoreLegend.png"  width='33%'alt="pLI score legend"><br>
  <li><b>Missense constraint track</b>: Missense constrained regions at the sub-genic level. All
    items are colored black.
</ol>
All tracks follow the general configuration settings for bigBed tracks. Mouseover on the LoF tracks
shows the pLI score and the upper bound of the O/E confidence interval for loss-of-function
variation (LOEUF), while mouseover on the missense track shows only the OE ratio. Clicking on items
in any track brings up a table of constraint metrics.
</p>

<p>
Clicking the grey box to the left of the track, or right-clicking and choosing the Configure option,
brings up the interface for filtering items based on their pLI/OE score, or labeling the items
based on their Ensembl identifier and/or Gene Name.
</p>

<H2>Methods</H2>
<p>
Please see the gnomAD browser help page and <a 
href="https://gnomad.broadinstitute.org/faq">FAQ</a> for further explanation of the topics below.</p>

<h3>Observed and Expected Variant Counts</h3>
<p>
<em>Observed count</em>: The number of unique single nucleotide variants in each transcript/gene
with 123 or fewer alternative alleles (MAF &lt; 0.1%).
</p>
<p>
<em>Expected count</em>: A depth corrected probability prediction model that takes into account
sequence context, coverage, and methylation, was used to predict expected
variant counts. For more information please see Lek et al. 2016.
</p>
<p>
Variants found in exons with median depth &lt; 1 were removed from both counts.
</p>

<h3>pLI and Z-scores</h3>
<p>
<em>NOTE</em>: The below information applies only to the Loss of Function tracks and not to the
missense constraint track, as these methods were developed after the publication of the
missense data.
</p>
<p>
The pLI and Z-scores are intended to measure how constrained or intolerant a gene or transcript is
to a specific type of variation. Genes or transcripts that are particularly depleted of a specific
class of variation (as observed in the gnomAD data set) are considered intolerant of that specific
type of variation. Z-scores are available for the Missense and Synonynmous categories and pLI
scores are available for the loss-of-function variation.
</p>
<p>
<em>Missense and Synonymous</em>: A signed Z-score of the deviation of observed counts relative to
the expected number. Positive Z-scores indicate more constraint (fewer observed variants than
expected), and negative scores indicate less constraint (more observed variants than expected). A
greater Z-score indicates more intolerance to the class of variation. Z-scores were generated by
a sequence-context based mutational model that predicted the number of expected rare (&lt; 1% MAF)
variants per transcript. The square root of the chi-squared value of the deviation of observed
counts from expected counts was multiplied by -1 if the observed count was greater than the expected
and vice versa. For the synonymous score, each Z-score was corrected by dividing by the standard
deviation of all synonymous Z-scores between -5 and 5. For the missense scores, a mirrored
distribution of all Z-scores between -5 and 0 was created, and then all missense Z-scores were
corrected by dividing by the standard deviation of the Z-score of the mirror distribution.
</p>
<p>
<em>Loss-of-function</em>: pLI closer to 1 indicates that the gene or transcript cannot tolerate
protein truncating variation (nonsense, splice acceptor and splice donor variation). The gnomAD
team recommends transcripts with a pLI of &gt;= 0.9 for the set of transcripts extremely intolerant
to truncating variants. pLI is based on the idea that transcripts can be classified into three
categories:
<ul>
  <li>null: heterozygous or homozygous protein truncating variation is completely tolerated
  <li>recessive: heterozygous variants are tolerated but homozygous variants are not
  <li>haploinsufficient: heterozygous variants are not tolerated
</ul>
An expectation-maximization algorithm was then used to assign a probability of belonging in each
class to each gene or transcript. pLI is the probability of belonging in the haploinsufficient class.
</p>

<p>
Please see Samocha et al 2014 and Lek et al 2016 for further discussion of these metrics.
</p>

<h3>Transcripts Included</h3>
<p>
Transcripts from GENCODE v19 were filtered according to the following criteria:
<ul>
  <li>Must have methionine at start of coding sequence
  <li>Must have stop codon at end of coding sequence
  <li>Must be divisible by 3
  <li>Must have at least one observed variant when removing exons with median depth &lt; 1
  <li>Must have reasonable number of missense and synonymous variants as determined by a Z-score cutoff
</ul>
After filtering the transcript set, 18225 transcripts were left over.
</p>


<h3>UCSC Track Methods</h3>
<h4>Loss of Function tracks</h4>
<p>
Per gene and per transcript data were downloaded from the gnomAD Google Storage bucket:
<pre>
gs://gnomad-public/release/2.1.1/constraint/gnomad.v2.1.1.lof_metrics.by_gene.txt.bgz
gs://gnomad-public/release/2.1.1/constraint/gnomad.v2.1.1.lof_metrics.by_transcript.txt.bgz
</pre>
These data were then joined to the Gencode v19 set of genes/transcripts available at the UCSC
Genome Browser and then transformed into a bigBed 12+5. For the full list of commands used to
make this track please see the &quot;gnomAD 2 pLI and other loss-of-function metrics&quot; section
of the
<a href="https://raw.githubusercontent.com/ucscGenomeBrowser/kent/master/src/hg/makeDb/doc/hg19.txt">makedoc</a>.
</p>
<h4>Missense Constraint track</h4>
<p>
Supplementary Table 4 from the <a href="https://www.biorxiv.org/content/10.1101/148353v1.full">
associated publication</a> was downloaded and joined to the Gencode v19 set of transcripts
available at UCSC and then transformed into a bigBed 12+6.  For the full list of commands
used to make this track please the &quot;gnomAD Missense Constraint Scores&quot; section of the
<a href="https://raw.githubusercontent.com/ucscGenomeBrowser/kent/master/src/hg/makeDb/doc/hg19.txt">makedoc</a>.
</p>


<h2>Data Access</h2>
<p>
The raw data can be explored interactively with the <a href="../hgTables">Table Browser</a>, or
the <a href="../hgIntegrator">Data Integrator</a>. For automated access, this track, like all 
others, is available via our <a href="../goldenPath/help/api.html">API</a>.  However, for bulk 
processing, it is recommended to download the dataset. The genome annotation is stored in a bigBed 
file that can be downloaded from the
<a href="http://hgdownload.soe.ucsc.edu/gbdb/hg19/gnomAD/pLI/">download server</a>. The exact
filenames can be found in the track configuration file. Annotations can be converted to ASCII text
by our tool <code>bigBedToBed</code> which can be compiled from the source code or downloaded as
a precompiled binary for your system. Instructions for downloading source code and binaries can be
found <a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">here</a>. The tool
can also be used to obtain only features within a given range, for example:</p>
<pre>
bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/hg19/gnomAD/pLI/pliByTranscript.bb -chrom=chr6 -start=0 -end=1000000 stdout
</pre>
<p>
Please refer to our
<A HREF="https://groups.google.com/a/soe.ucsc.edu/forum/?hl=en&fromgroups#!search/gnomAD"
target=_blank>mailing list archives</a>
for questions and example queries, or our
<a HREF="../FAQ/FAQdownloads.html#download36" target=_blank>Data Access FAQ</a>
for more information.
</p>

<H2>Credits</H2>
<P>
Thanks to the gnomAD team, especially Anne O'Donnell Luria and Emily O'Heir for their help with
making these data and recommendations on how to display the data.
</P>

<H2>References</H2>
<p>
Karczewski KJ, Francioli LC, Tiao G, Cummings BB, Alfoldi J, Wang Q, Collins RL, Laricchia KM, Gann A, Birnbaum DP <em>et al</em>.
<a href="https://doi.org/10.1101/531210 " target="_blank">
Variation across 141,456 human exomes and genomes reveals the spectrum of loss-of-function intolerance across human protein-coding genes</a>.
<em>bioRxiv</em>. 2019 August 13.
<a href="https://www.biorxiv.org/content/10.1101/531210v3" target="_blank">bioRxiv</a>
</p>

<p>
Lek M, Karczewski KJ, Minikel EV, Samocha KE, Banks E, Fennell T, O'Donnell-Luria AH, Ware JS, Hill
AJ, Cummings BB <em>et al</em>.
<a href="https://doi.org/10.1038/nature19057" target="_blank">
Analysis of protein-coding genetic variation in 60,706 humans</a>.
<em>Nature</em>. 2016 Aug 18;536(7616):285-91.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27535533" target="_blank">27535533</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5018207/" target="_blank">PMC5018207</a>
</p>

<p>
Samocha KE, Robinson EB, Sanders SJ, Stevens C, Sabo A, McGrath LM, Kosmicki JA, Rehnstr&#246;m K,
Mallick S, Kirby A <em>et al</em>.
<a href="http://dx.doi.org/10.1038/ng.3050" target="_blank">
A framework for the interpretation of de novo mutation in human disease</a>.
<em>Nat Genet</em>. 2014 Sep;46(9):944-50.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25086666" target="_blank">25086666</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4222185/" target="_blank">PMC4222185</a>
</p>

<p>
Samocha KE, Kosmicki JA, Karczewski KJ, O'Donnell-Luria AH, Pierce-Hoffman E, MacArthur DG, Neale BM, Daly MJ.
<a href="https://doi.org/10.1101/148353 " target="_blank">
Regional missense constraint improves variant deleteriousness prediction</a>.
<em>bioRxiv</em>. 2017 June 12.
<a href="https://www.biorxiv.org/content/10.1101/148353v1.full" target="_blank">bioRxiv</a>
</p>
