<H2>Description</H2>
<P>
The gnomAD predicted loss-of-function (pLoF) track set shows per gene and per transcript loss of
function constraint metrics as predicted by the
<a href="https://gnomad.broadinstitute.org">Genome Aggregation Database (gnomAD)</a> for gnomAD v2.1.1.
The metrics include:
<ul>
  <li>observed and expected variant counts per transcript/gene
  <li>Observed/Expected  ratio (O/E)
  <li>Z-scores of the observed counts compared to expected, for missense and synonymous categories only
  <li>probability loss of function intolerant (pLI), for loss-of-function variation only
</ul>
The O/E ratio is a continuous measurement of how tolerant a gene or transcript is to a certain
class of variation. Each item in this track shows the O/E ratio for three different types of
variation: missense, synonymous, and loss-of-function. The confidence interval is displayed along
with the O/E ratio to better assist interpration of the scores. Please see the Methods section
below for more information about how the scores were calculated.
</p>

<h2>Display Conventions and Configuration</h2>
<p>
Both tracks follow the general configuration settings for bigBed tracks. Transcripts or genes are
shaded according to their pLI score, and the legend below the track image describes the color
gradient the scores fall into. Mouseover on each item shows the pLI score and the upper bound of
the O/E confidence interval for loss-of-function variation (LEOUF). Clicking on an item in
either track shows a table of the predicted Missense/Synonymous/pLoF metrics.
</p>

<p>
Clicking the grey box to the left of the track, or right-clicking and choosing the Configure option
brings up the interface for filtering items based on their pLI score, or labeling the items
based on their Ensembl identifier and/or Gene Name.
</p>

<H2>Methods</H2>
<p>
Please see the gnomAD browser help page and <a href="https://gnomad.broadinstitute.org/faq">FAQ</a> for further explanation of the below topics.
<h3>Observed and Expected Variant Counts</h3>
<p>
<em>Observed count</em>: The number of unique single nucleotide variants in each transcript/gene
with 123 or fewer alternative alleles (MAF &lt; 0.1%).
</p>
<p>
<em>Expected count</em>: A depth corrected probability prediction model was used to predict expected
variant counts. For more information please see Lek et al. 2016.
</p>
<p>
Variants found in exons with median depth &lt; 1 were removed from both counts.
</p>

<h3>pLI and Z-scores</h3>
<p>
The pLI and Z-scores are intended to measure how constrained or intolerant a gene or transcript is
to a specific type of variation. Genes or transcripts that are particularly depleted of a specifc
class of variation (as observed in the gnomAD data set) are considered intolerant of that specific
type of variation. Z-scores are available for the Missense and Synonynmous categories and pLI
scores are available for the loss-of-function variation.
</p>
<p>
<em>Missense and Synonymous</em>: A signed Z-score of the deviation of observed counts relative to
the expected number. Positive Z-scores indicate more constraint (less observed variants than
expected), and negative scores indicate less constraint (more observed variants than expected). A
greater Z-score indicates more intolerance to the class of variation. Z-scores were generated by
a sequence-context based mutational model that predicted the number of expected rare (&lt; 1% MAF)
variants per transcript. The square root of the chi-squared value of the deviation of observed
counts from expected counts was multiplied by -1 if the observed count was greater than the expected
and vice versa. For the synonymous score, each Z-score was corrected by dividing by the standard
deviation of all synonynous Z-scores between -5 and 5. For the missense scores, a mirrored
distribution of all Z-scores between -5 and 0 was created, and then all missense Z-scores were
corrected by dividing by the standard deviation of the Z-score of the mirror distribution.
</p>
<p>
<em>Loss-of-function</em>: pLI closer to 1 indicates that the gene or transcript cannot tolerate
protein truncating variation (nonsense, splice acceptor and splice donor variation). The gnomAD
team recommends transcripts with a pLI of &gt;= 0.9 for the set of transcripts extremely intolerant
to truncating variants. pLI is based on the idea that transcripts can be classified into three
categories:
<ul>
  <li>null: heterozygous or homozygous protein truncating variation is completely tolerated
  <li>recessive: heterozygous variants are tolerated but homozygous variants are not
  <li>haploinsufficient: heteroyzgous variants are not tolerated
</ul>
An expectation-maximization algorithm was then used to assign a probability of belonging in each
class to each gene or transcript. pLI is the probability of belonging in the haploinsufficient class.
</p>

<p>
Please see Samocha et al 2014 and Lek et al 2016 for further discussion of these metrics.
</p>

<h3>Transcripts Included</h3>
<p>
Transcripts from GENCODE v19 were filtered according to the following criteria:
<ul>
  <li>Must have methionine at start of coding sequence
  <li>Must have stop codon at end of coding sequence
  <li>Must be divisible by 3
  <li>Must have at least one observed variant when removing exons with median depth &lt; 1
  <li>Must have reasonable number of missense and synonymous variants as determined by a Z-score cutoff
</ul>
After filtering the transcript set, 18225 transcripts were left over.
</p>

<h3>UCSC Track Methods</h3>
<p>
Per gene and per transcript data were downloaded from the gnomAD Google Storage bucket:
<pre>
gs://gnomad-public/release/2.1.1/constraint/gnomad.v2.1.1.lof_metrics.by_gene.txt.bgz
gs://gnomad-public/release/2.1.1/constraint/gnomad.v2.1.1.lof_metrics.by_transcript.txt.bgz
</pre>
These data were then joined to the Gencode v19 set of genes/transcripts available at the UCSC
Genome Browser and then transformed into a bigBed 12+5. For the full list of commands used to
make this track please see the &quot;gnomAD 2 pLI and other loss-of-function metrics&quot; section
of the
<a href="https://raw.githubusercontent.com/ucscGenomeBrowser/kent/master/src/hg/makeDb/doc/hg19.txt">makedoc</a>.
</p>

<h2>Data Access</h2>
<p>
The raw data can be explored interactively with the <a href="../hgTables">Table Browser</a>, or
the <a href="../hgIntegrator">Data Integrator</a>. For automated analysis, the genome annotation
is stored in a bigBed file that can be downloaded from the
<a href="http://hgdownload.soe.ucsc.edu/gbdb/hg19/gnomAD/pLI/">download server</a>. The exact
filenames can be found in the track configuration file. Annotations can be converted to ASCII text
by our tool <code>bigBedToBed</code> which can be compiled from the source code or downloaded as
a precompiled binary for your system. Instructions for downloading source code and binaries can be
found <a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">here</a>. The tool
can also be used to obtain only features within a given range, for example:
<pre>
bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/hg19/gnomAD/pLI/pliByTranscript.bb -chrom=chr6 -start=0 -end=1000000 stdout
</pre>
Please refer to our mailing list archives for questions, or our Data Access FAQ for more information. 
</p>

<H2>Credits</H2>
<P>
Thanks to the gnomAD team, especially Anne O'Donnell Luria and Emily O'Heir for their help with
making these data and recommendations on how to display the data.
</P>

<H2>References</H2>

<p>
</p>
