<H2>Description</H2>
<P>
The gnomAD predicted loss of function (pLoF) track set shows per gene and per transcript loss of
function constraint metrics as predicted by the
<a href="gnomad.broadinstitute.org">Genome Aggregation Database (gnomAD)</a>.
For each gene or transcript, and for each
constraint category (missense, synonymous, loss of function), the observed/expected counts, the
(O/E) ratio and confidence interval, and either the Z-score of the O/E value for the missense and
synonymous categories, or the predicted loss of function intolerance score for the loss of function
category. In addition, the upper bound of the O/E ratio for the loss of function category describes
the loss of function observed/expected ratio upper bound fraction (LOEUF) score.
</p>

<h2>Display Conventions and Configuration</h2>
<p>
Both tracks follow the general configuration settings for bigBed tracks. Transcripts or genes are
shaded according to their pLI score, and the legend below the track image describes the color
gradient the scores fall into. Mouseover on each item shows the pLI score and the LOEUF score.
Clicking on an item in either track shows a table of the predicted Missense/Synonymous/pLoF metrics.
</p>

<p>
Clicking the grey box to the left of the track, or right-clicking and choosing the Configure option
brings up the interface for filtering items based on their pLI score, or labeling the items
based on their Ensembl identifier and/or Gene Name.
</p>

<H2>Methods</H2>
<h3>Transcripts Included</h3>
<P>
Transcripts from GENCODE v19 were filtered according to the following criteria:
<ul>
  <li>Must have methionine at start of coding sequence
  <li>Must have stop codon at end of coding sequence
  <li>Must be divisible by 3
  <li>Must have at least one observed variant when removing exons with median depth &lt; 1
  <li>Must have reasonable number of missense and synonymous variants as determined by a Z-score cutoff
</ul>
After filtering the transcript set, 18225 transcripts were left over.
</p>

<h3>Observed and Expected Variant Counts</h3>
<h3>pLI and O/E ratio</h3>

<h3>UCSC Track Methods</h3>
<P>
Per gene and per transcript data were downloaded from the gnomAD Google Storage bucket:
<pre>
gs://gnomad-public/release/2.1.1/constraint/gnomad.v2.1.1.lof_metrics.by_gene.txt.bgz
gs://gnomad-public/release/2.1.1/constraint/gnomad.v2.1.1.lof_metrics.by_transcript.txt.bgz
</pre>
These data were then joined to the Gencode v19 set of genes/transcripts available at the UCSC
Genome Browser and then transformed into a bigBed 12+5. For the full list of commands used to
make this track please see the &quot;gnomAD 2 pLI and other loss of function metrics&quot; section
of the
<a href="https://raw.githubusercontent.com/ucscGenomeBrowser/kent/master/src/hg/makeDb/doc/hg19.txt">makedoc</a>.
</P>

<h2>Data Access</h2>
<p>
</p>

<H2>Credits</H2>
<P>
Thanks to the gnomAD team, especially Anne O'Donnell Luria and Emily O'Heir for their help with
making these data and recommendations on how to display the data.
</P>

<H2>References</H2>

<p>
</p>
