<h2>Description</h2>
<p>
The <b>${longLabel}</b> track set shows isoform expression values across tissues. The pext metric 
allows for quick visualization of the expression status of exons across a gene. The median 
transcript expression levels in 53 tissue samples was calculated based on RNA-seq data from the 
GTEx v7 dataset.  
</p>
<p>
The <b>gnomAD PEXT mean_proportion</b> subtrack displays the mean expression of a transcript calculated 
from all 53 GTEx tissue samples. 
</p>
<p>
Note that isoform quantification tools can be imprecise, especially for longer genes with many 
annotated isoforms. Regions with low pext values might be enriched for annotation errors (ie. there 
may be edge cases for which an exon that is established to be critical for gene function may appear 
unexpressed with pext). Also note that the GTEx dataset is postmortem adult tissue, and thus 
the possibility that an exon may be development-specific or may be expressed in tissues not 
represented in GTEx can not be dismissed.
</p>

<h2>Display Conventions and Configuration</h2>

<p>
The pext graphs display the mean expression at each base position for protein-coding (CDS) regions.
While UTRs do have expression in transcriptome datasets, note that this information is not included 
for the visualization. The details page shows calculated sample percentages for the range of 
sequence within the browser window. 
</p>

<H2>Methods</H2>

<p>
The pext values are derived from isoform quantifications using the <a target="blank"
href="http://deweylab.biostat.wisc.edu/rsem/">RSEM tool</a>. Detailed information about 
development and commands to create these files can be found <a target="blank" 
href="https://github.com/macarthur-lab/tx_annotation/">here</a>. 
</p>

<h2>Data Access</h2>
<p>
The raw data can be explored interactively with the <a href="../hgTables">Table Browser</a>, or
the <a href="../hgIntegrator">Data Integrator</a>. For automated access, this track, like all 
others, is available via our <a href="../goldenPath/help/api.html">API</a>.  
The data can also be found directly from the gnomAD <a target="_blank"
href="https://gnomad.broadinstitute.org/downloads">downloads page</a>. 
</p>

<p>
Please refer to our
<A HREF="https://groups.google.com/a/soe.ucsc.edu/forum/?hl=en&fromgroups#!search/gnomAD"
target="_blank">mailing list archives</a>
for questions and example queries, or our
<a HREF="../FAQ/FAQdownloads.html#download36" target="_blank">Data Access FAQ</a>
for more information.</p>

<h2>Credits</h2>
<p>
Thanks to the gnomAD team, especially Anne O'Donnell Luria and Emily O'Heir for their help with
making these data and recommendations on how to display the data.
</p>

<h2>References</h2>
<p>
Cummings BB, Karczewski KJ, Kosmicki JA, Seaby EG, Watts NA, Singer-Berk M, Mudge JM, Karjalainen J,
Satterstrom FK, O'Donnell-Luria AH <em>et al</em>.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/32461655" target="_blank">
Transcript expression-aware annotation improves rare variant interpretation</a>.
<em>Nature</em>. 2020 May;581(7809):452-458.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32461655" target="_blank">32461655</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7334198/" target="_blank">PMC7334198</a>
</p>
