<h2>Description</h2>
<p>
This track shows most variants found in the full text of scientific publications gathered by
<a target=_blank href="https://www.genomenon.com/">Genomenon Mastermind</a>. Mastermind
uses a software that searches for disease-gene-variant associations in the 
scientific literature. The genome browser track shows only if a
variant has been indexed by the search engine.
</p>

<p>
To get details on a variant (bibliographic references, disease, etc)
click it and follow the &quot;Protein change and link to details&quot; at the top
of the details page. Mouse over an item to show the gene and amino acid change and the 
scores MMCNT1, MMCNT2 and MMCNT3, explained below.
</p>

<p>
Genomenon Mastermind Genomic Search Engine is a commercial database of variants
likely to be mentioned in full text scientific articles. A limited number of
queries per week is free for healthcare professionals and researchers, if they register on the
<a href="https://mastermind.genomenon.com/users/sign_up/" target=_"blank">signup
page</a> page. Advanced features require a license for the
<a href="https://www.genomenon.com/mastermind-pro-upgrade/" target="_blank">Mastermind Professional Edition</a>, 
which contains the same content but allows more comprehensive searches.
</p>

<h2>Display Conventions and Configuration</h2>

<p>
Genomic locations of variants are labeled with the nucleotide change.
Hover over the features to see the gene, the amino acid change and the scores MMCNT1, MMCNT2 and 
MMCNT3, described below. All other information is shown on the respective Mastermind variant detail
page, accessible via the &quot;Link to Mastermind&quot; at the top of the details page. The feature
are colored based on their
evidence:
</p>

<p>
<table>
  <thead>
  <tr>
    <th style="border-bottom: 2px solid #6678B1;">Color</th>
    <th style="border-bottom: 2px solid #6678B1;">Level of support</th>
  </tr>
  </thead>
  <tr>
    <th bgcolor="#0C0C78"></th>
    <th align="left">High: at least one paper mentions this exact cDNA change</th>
  </tr>
  <tr>
    <th bgcolor="#5050A0"></th>
    <th align="left">Medium: at least two paper mention a variant that leads to the same amino acid change</th>
  </tr>
  <tr>
    <th bgcolor="#8282D2"></th>
    <th align="left">Low: only a single paper mentions a variant that leads to the same amino acid change</th>
  </tr>
</table>
</p>

<p>
The three numbers that are shown on the mouse over and the details page have the following meaning (MM=Mastermind):
<ul>
    <li><b>MMCNT1</b>: cDNA-level exact matches. This is the number of articles that mention the variant at the nucleotide level in either the title/abstract or the full-text.
    <li><b>MMCNT2</b>: cDNA-level possible matches. This is the number of articles with nucleotide-level matches (from 1) plus articles with protein-level matches in which the publication did not specify the cDNA-level change, meaning they could be referring to this nucleotide-level variant but there is insufficient data in these articles to determine conclusively.
    <li><b>MMCNT3</b>: This is the number of articles citing any variant resulting in the same biological effect as this variant. This includes the articles from MMCNT1 and MMCNT2 plus articles with alternative cDNA-level variants that result in the same protein effect.
</ul>

</p>

<h2>Methods</h2>

The Mastermind Cited Variants file was <a target=_blank href="https://www.genomenon.com/cvr/">downloaded</a>,
converted to BED format with scripts that are available in our 
<a target=_blank href="https://github.com/ucscGenomeBrowser/kent/tree/master/src/hg/makeDb/mastermind/">Git
repository</a> and converted to a bigBed file with the UCSC genome browser tool
bedToBigBed.

<h2>Credits</h2>

<p>
Thanks to Mark Kiel, Steve Schwartz and Clayton Wheeler from Genomenon for making this data available.
</p>

<!-- <h2>References</h2> -->

