<h2>Description</h2>
<p>
This track contains mappings of single nucleotide variants
and small insertions and deletions (indels) &mdash; collectively Simple
Nucleotide Variants (SNVs) &mdash; from the European Variation Archive
(<a href="https://www.ebi.ac.uk/eva/" target="_blank">EVA</A>)
Release 3 for the $organism $db genome. The dbSNP database at NCBI no longer
hosts non-human variants.

</p>

<h2>Interpreting and Configuring the Graphical Display</h2>
<p>
  Variants are shown as single tick marks at most zoom levels.
  When viewing the track at or near base-level resolution, the displayed
  width of the SNP variant corresponds to the width of the variant in the
  reference sequence. Insertions are indicated by a single tick mark displayed
  between two nucleotides, single nucleotide polymorphisms are displayed as the
  width of a single base, and multiple nucleotide variants are represented by a
  block that spans two or more bases.
</p>

<p>
  Navigation to an individual variant can be accomplished by typing or copying
  the variant identifier (rsID) or the genomic coordinates into the Position/Search box on the Browser.
<p>
  Variants are colored according to functional effect on genes annotated by dbSNP:
</p>

<p>
<table cellpadding='2'>
  <thead><tr>
    <th style="border-bottom: 2px solid;">Color</th>
    <th style="border-bottom: 2px solid;">Variant Type</th>
  </tr></thead>
  <tr><td style="background-color: red"></td><td>Protein-altering variants and splice site variants</td></tr>
  <tr><td style="background-color: green"></td><td>Synonymous codon variants</td></tr>
  <tr><td style="background-color: blue"></td><td>Non-coding transcript or Untranslated Region (UTR) variants</td></tr>
  <tr><td style="background-color: black"></td><td>Intergenic and intronic variants</td></tr>
</table>
</p>

<p>
  A click on an item in the graphical display displays a page with data about
  that variant.  Data fields include the Reference and Alternate Alleles, the
  class of the variant as reported by EVA, the source of the data, the amino acid
  change, if any, and the functional class as determined by UCSC's Variant Annotation
  Integrator.
</p>

</p>
  Variants are classified by EVA into one of the following classes:
</p>

<ul>
  <li> <b>substitution</b> &mdash;
       A single nucleotide in the reference is replaced by another, alternate allele
  <li> <b>deletion</b> &mdash; 
       One or more nucleotides is deleted.  The representation in the database is to
       display one additional nucleotide in both the Reference field (Ref) and the 
       Alternate Allele field (Alt).  E.g. a variant that is a deletion of an A
       maybe be represented as Ref = GA and Alt = G.
  <li> <b>insertion</b> &mdash; 
       One or more nucleotides is inserted.  The representation in the database is to
       display one additional nucleotide in both the Reference field (Ref) and the 
       Alternate Allele field (Alt).  E.g. a variant that is an insertion of a T maybe 
       be represented as Ref = G and Alt = GT 
  <li> <b>tandemRepeat</b> &mdash; 
       The Reference and Alternate Alleles are runs of the same nucleotide,
       but of different length, e.g., Ref = CCCC; Alt = CCCCCC 
  <li> <b>delins</b> &mdash; 
       Similar to tandemRepeat, in that the runs of Ref and Alt Alleles are of
       different length, except that there is more than one type of nucleotide,
       e.g., Ref = CCAAAAACAAAAACA, Alt = ACAAAAAC.
  <li> <b>multipleNucleotideSubstitution</b> &mdash; 
       More than one nucleotide is substituted by an equal number of different 
       nucleotides, e.g.,  Ref = AA, Alt = GC.
</ul>
</p>

</p>
  Variants can be filtered using the track controls to show subsets of the
  data by either EVA class or UCSC-generated functional annotation.
</p>

<h2>Methods</h2>
<p>
Data were downloaded from the European Variation Archive as a
<a href="https://ftp.ebi.ac.uk/pub/databases/eva/rs_releases/release_3/by_species/canis_lupus_familiaris/GCA_000002285.2/GCA_000002285.2_current_ids.vcf.gz"
target="_blank">vcf.gz</a> file
(release date, 2022-02-24) and converted to BED and then bigBed format.
Sequence Ontology (&quot;<a href="http://www.sequenceontology.org/browser/current_release"
target="_blank">SO</a>:&quot;)
terms were converted to the variant classes described above.
</p>
<p>
No functional annotations were provided by the EVA (e.g., missense, nonsense, etc).
These were computed using UCSC's Variant Annotation Integrator (Hinrichs, et al., 2016).
</p>

<h2>Data Access</h2>
<p>
For automated download and analysis, this annotation is stored in a bigBed file that
can be downloaded from <a href="https://hgdownload.soe.ucsc.edu/gbdb/$db/bbi/"
target="_blank">our download server</a>. The file for this track is called <tt>evaSnp.bb</tt>.
Individual regions or the whole genome annotation can be obtained using our tool
<tt>bigBedToBed</tt> which can be compiled from the source code or downloaded as a precompiled
binary for your system. Instructions for downloading source code and binaries can be found
<a href="https://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">here</a>.
The tool can also be used to obtain only features within a given range, e.g.
<br><br>
<tt>bigBedToBed https://hgdownload.soe.ucsc.edu/gbdb/$db/encode3/evaSnp.bb -chrom=chr21 -start=0 -end=100000000 stdout</tt>
</p>

<h2>Credits</h2>
<p>
This track was produced at UCSC by Robert Kuhn from data generated worldwide.
</p>

<h2>References</h2>
<p>
Cezard T, Cunningham F, Hunt SE, Koylass B, Kumar N, Saunders G, Shen A, Silva AF,
Tsukanov K, Venkataraman S <em>et al.</em> <a href="https://doi.org/10.1093/nar/gkab960"
target="_blank">The European Variation Archive: a FAIR resource of genomic variation for all
species</a>.  <em>Nucleic Acids Res.</em> 2021 Oct 28:gkab960.
<a href="https://doi.org/10.1093/nar/gkab960" target="_blank">doi:10.1093/nar/gkab960</a>.
Epub ahead of print. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/34718739/"
target="_blank">34718739</a>. PMID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/pmc8728205/"
target="_blank">PMC8728205</a>.
</p>
<p>
Hinrichs AS, Raney BJ, Speir ML, Rhead B, Casper J, Karolchik D, Kuhn RM, Rosenbloom KR, Zweig AS,
Haussler D, Kent WJ.
<a href="https://academic.oup.com/bioinformatics/article/32/9/1430/1744314/"
target="_blank">UCSC Data Integrator and Variant Annotation Integrator</a>.
<em>Bioinformatics</em>. 2016 May 1;32(9):1430-2.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26740527" target="_blank">26740527</a>; PMC:
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4848401/" target="_blank">PMC4848401</a>
</p>
