<H2>Description</H2>
<P>
This group of tracks shows data from <a href="https://science.sciencemag.org/content/370/6518/eaba7721"><i>A human cell atlas of fetal gene expression</i></a>. This is a collection of single cell and single nucleus droplet-based RNA-seq data covering 4 million cells from 15 organs obtained during mid-gestation.  The cells were sequenced in a highly multiplexed fashion, and then clustered and the clusters were annotated as described in the paper. 

The Fetal Cells subtrack contains the data organized by cell type, with RNA signals from all cells of a given type pooled and averaged into one bar for each cell type.  The Fetal lineage shows similar data, but with the cell types subdivided more finely and by organ.  Additional bar chart subtracks pool the cell by other characterists such as by sex.

Please see <a href='https://descartes.brotmanbaty.org/'>descartes.brotmanbaty.org</a> for further interactive displays and additional data.

<H2>Display Conventions and Configuration</H2>
<p>
The cell types are colored by which class they belong to according to the following table.
The coloring algorithm allows cells that show some blended characteristics to show blended
colors so there will be some color variation within a class.  The colors will be purest in
the Fetal Cells subtrack, where the bars represent relatively pure cell types.  They can
give an overview of the cell composition within other categories in other subtracks as well.
<p>
<table cellpadding='2'>
  <thead><tr>
    <th style="border-bottom: 2px solid;">Color</th>
    <th style="border-bottom: 2px solid;">Cell classification</th>
  </tr></thead>
  <tr><td style='background-color: #F0C000;'></td><td>neural</td></tr>
  <tr><td style="background-color: #FF8838;"></td><td>adipose</td></tr>
  <tr><td style="background-color: #B05020;"></td><td>fibroblast</td></tr>
  <tr><td style="background-color: #F01111;"></td><td>immune</td></tr>
  <tr><td style="background-color: #C000C0;"></td><td>muscle</td></tr>
  <tr><td style="background-color: #6000B0;"></td><td>hepatocyte</td></tr>
  <tr><td style="background-color: #5020E0;"></td><td>trophoblast</td></tr>
  <tr><td style="background-color: #0050FF;"></td><td>secretory</td></tr>
  <tr><td style="background-color: #0072D6;"></td><td>ciliated</td></tr>
  <tr><td style="background-color: #0099C0;"></td><td>epithelial</td></tr>
  <tr><td style="background-color: #00C000;"></td><td>endothelial</td></tr>
  <tr><td style="background-color: #80D000;"></td><td>glia</td></tr>
</table>


<H2>Methods</H2>
<p>
Three-level single-cell combinatorial indexing (sci-RNAseq3) as described in (ref) was used 
on 121 samples from 28 fetuses estimated 72 to 129 days post-conception.  This included samples
from 15 organs. and resulted in RNA profiles for 4 million cells. The samples were flash-frozen
for majority of the experiments and then nuclei extracted for sequencing. Samples from tissues 
from the kidney and digestive system were fixed after disassociation to deactivate endogenous
RNases and proteases
</p>
<p>The cell/gene matrix and cell-level metadata was downloaded from the <a href="https://cells-test.gi.ucsc.edu/?ds=fetal-gene-atlas+genes+all">UCSC Cell Browser(fix URL)</A>.  The UCSC command line tool matrixClusterColumns, matrixToBarChart,
and bedToBigBed were used to transform these into a bar chart format bigBed file that can be
visualized.  The coloring was done by defining colors for the broad level cell classes and then
using another UCSC tool, hcaColorCells, to interpolate the colors across all cell types.

<H2>Credits</H2>
<p>Thanks to the many authors who worked on producing and publishing this data set.  
The data was integrated into the UCSC Genome Browser by Jim Kent and QA'd by ???
The UCSC work was paid for by the Chan Zuckerberg Initiative.
</p>

<H2>References</H2>
<UL>
<LI>Junyue Cao, Diana R. O'Day, Hannah A. Pliner, Paul D. Kingsley, Mei Deng, Riza M. Daza,
Michael A. Zager, Kimberly A. Aldinger, Ronnie Blecher-Gonen, Fan Zhang, Malte Spielmann,
James Palis, Dan Doherty, Frank J. Steemers, Ian A. Glass, Cole Trapnell, Jay Shendure
<a href='https://science.sciencemag.org/content/370/6518/eaba7721.full'>A human cell atlas of fetal gene expression</a> Science 2020
<LI> J. Cao, M. Spielmann, X. Qiu, X. Huang, D. M. Ibrahim, A. J. Hill, F. Zhang, S. Mundlos, L. Christiansen, F. J. Steemers, C. Trapnell, J. Shendure, 
<a href='https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6434952/'>The single-cell transcriptional landscape of mammalian organogenesis.</a> Nature 2019
</UL>>

<!-- seize control of background color -->
<style>
body.cgi {
   background: #F0F0F0;
}
table.hgInside {
   background: #FFFFFF;
}
</style>

