<h2>Description</h2>

<p>
This track shows the genomic positions of somatic variants found through whole genome sequencing of tumors
as part of The Cancer Genome Atlas (TCGA) by
the National Cancer Institute, made 
available through the <a href="https://portal.gdc.cancer.gov" target="_blank">Genomic Data Commons Portal</a>.
The data shown here is sometimes called the "Pan-Cancer dataset", a collection
of thirty-three TCGA projects processed in a uniform way. 

<h2>Display Conventions and Configuration</h2>
<p>The variants can be shown in two display modes:
<ol>
<li>Lollipop Mode: variants are shown with circles, the y-axis represents the
number of samples within the cohort. This mode is active if the track is in full or squish mode.</li>
<li>BED Mode: variants are shown as normal genomic features. This mode is active if the track is in pack or dense mode.</li>
</ol>
</p>

<p>Variants can be filtered by project and gender from the track details page. Pressing the "Advanced" button allows the user to specify whether the checked values all have to be true of a particular variant, or if only one of them need be present to satisfy the filter.</p>

<p>The vertical viewing range in full mode can also be used to filter what variants are shown. Variants that have a sampleCount 
more or less than the min and max values specificed in the viewing range are not displayed</p>

<H2>Data access</H2>
<p>
The raw data can be explored interactively with the <a href="../cgi-bin/hgTables">Table Browser</a>
or the <a href="../cgi-bin/hgIntegrator">Data Integrator</a>.

<p>
For automated download and analysis, the genome annotation for all the thirty-three projects is stored in a bigBed file that
can be downloaded from
<a href="http://hgdownload.soe.ucsc.edu/gbdb/hg38/gdcCancer/gdcCancer.bb" target="_blank">our download server</a>.
There are also bigBed files for each of the thirty-three projects in that directory.
 Individual
regions or the whole genome annotation can be obtained using our tool <tt>bigBedToBed</tt>
which can be compiled from the source code or downloaded as a precompiled
binary for your system. Instructions for downloading source code and binaries can be found
<a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">here</a>.
The tool
can also be used to obtain only features within a given range, e.g. 
<tt>bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/hg38/gdcCancer/gdcCancer.bb -chrom=chr21 -start=0 -end=100000000 stdout</tt></p>
</p>


<h2>Methods</h2>

<p>
All Mutect Variant calls were downloaded from the GDC portal in January 2019 and
reformatted at UCSC to the <a href="../goldenPath/help/bigBed.html">bigBed</a>
format with a short script, 
<a href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/utils/cancerMafToBigBed">cancerMafToBigBed</a>.
</p>

<h2>Credits</h2>
<p>
Thanks to GDC for making the TCGA data available on their web site.
</p>
