<h2>Description</h2>
<p>
MANE, Matched Annotation from the NCBI and Ensembl-EBI, provides a matched set of well-supported
transcripts for human protein-coding genes and defines one representative transcript for each
gene. This data is intended to be a common reference transcript dataset, including one or a few key
transcripts for each gene where RefSeq and Ensembl/GENCODE transcripts are identical in length
and sequence, and completely match the human reference genome sequence. 

To pick representative transcripts, the MANE collaborators developed computational methods to 
evaluate transcript expression levels, protein conservation, support from archived transcript 
submissions, clinical relevance, and other factors.  Complex genes are subject to review by
annotation experts from both groups and often make improvements to both annotation sets.

This minimal, identically annotated transcript set will eliminate the need to choose between RefSeq
and Ensembl/GENCODE datasets for genomic analyses. This should make it easy for researchers who
currently prefer one dataset over the other to exchange data or translate coordinates
(or HGVS variation expressions) between RefSeq and Ensembl annotation results.

Note that this data (from MANE 0.5) is a beta release and only covers 53% or protein coding genes.
Both RefSeq and Ensembl will continue to provide a rich set of alternate transcripts
in the RefSeq-All and the GENCODE tracks.

<h2>Data Access</h2>
<p>
The raw data can be explored interactively with the <a href="../cgi-bin/hgTables"
target="_blank">Table Browser</a>, or the <a href="../cgi-bin/hgIntegrator"
target="_blank">Data Integrator</a>. For computational analysis, genome annotations are stored in
a bigGenePred file that can be downloaded from the
<a href="http://hgdownload.soe.ucsc.edu/gbdb/$db/mane/" target="_blank">download
server</a>. Regional or genome-wide annotations can be converted from binary data to human readable
text using our command line utility <em>bigBedToBed</em> which can be compiled from source code or
downloaded as a precompiled binary for your system. Instructions for download can be found in the
<a href="http://hgdownload.soe.ucsc.edu/admin/exe/" target="_blank">utilities directory</a>.
The utility can be used to obtain features within a given range, for example:</p>
<code>bigBedToBed -chrom=chr6 -start=0 -end=1000000 http://hgdownload.soe.ucsc.edu/gbdb/hg38/mane/mane.0.5.bb stdout</code>
<p>
Please refer to our <a href="../FAQ/FAQdownloads.html#download36" target="_blank">Data Access FAQ
</a> for more information or our
<a href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome"
target="_blank">mailing list</a> for archived user questions.</p>

<h2>Credits</h2>
<p>
Thank you to the RefSeq project at NCBI and the Ensembl/GENCODE project at EMBL-EBI.
You can contact the authors directly at MANE-help@ncbi.nlm.nih.gov or
mane-help@ebi.ac.uk.

<h2>References</h2>
Release announcement by NCBI: <a href="https://ncbiinsights.ncbi.nlm.nih.gov/2019/03/12/mane-select-v0-5">https://ncbiinsights.ncbi.nlm.nih.gov/2019/03/12/mane-select-v0-5</a> <br>
Download links for MANE Select v0.5: <a href="ftp://ftp.ncbi.nlm.nih.gov/refseq/MANE">ftp://ftp.ncbi.nlm.nih.gov/refseq/MANE</a><br>
<p>
