<h2>Description</h2>

<div class="warn-note" style="border: 2px solid #9e5900; padding: 5px 20px; background-color: #ffe9cc;">
<p><span style="font-weight: bold; color: #c70000;">NOTE:</span><br> 
ClinVar is intended for use primarily by physicians and other
professionals concerned with genetic disorders, by genetics researchers, and
by advanced students in science and medicine. While the ClinVar database is
open to all academic users, users seeking information about a personal medical
or genetic condition are urged to consult with a qualified physician for
diagnosis and for answers to personal questions.</p>
</div>

<p>
These tracks show the genomic positions of variants in the
<a href="https://www.ncbi.nlm.nih.gov/clinvar/" target="_blank">ClinVar database</a>. 
ClinVar is a free, public archive of reports
of the relationships among human variations and phenotypes, with supporting
evidence. </p>

<p>
The <b>ClinVar CNVs track</b> displays copy number variants 
(CNVs) equal or larger than 50 bp, and the <b>ClinVar SNVs track</b> displays substitutions and 
indels shorter than 50 bp. 
Until October 2017, all variants with the ClinVar types 
<em>copy number gain/loss</em> and <em>DbVar "nsv" accessions</em> were assigned in the CNV 
category. Because the ClinVar type no longer captures this information, any variation equal or 
larger than 50 bp is now considered a CNV.
</p>

<p>
<b>Note:</b> The data in the track are obtained directly from ClinVar's FTP site.
We display the data obtained from ClinVar as-is to avoid discrepancies between UCSC and NCBI. 
However, be aware that the ClinVar conventions are different from the VCF standard. 
Variants may be right-aligned or may contain additional context, e.g. for
inserts. ExAC/gnomAD make available <a href="https://github.com/macarthur-lab/clinvar"
target="_blank">a converter</a>
to make ClinVar more comparable to VCF files.</p>

<h2>Display Conventions and Configuration</h2>

<p>
The genomic locations of ClinVar variants are labelled with the ClinVar variant descriptions. 
<b>Mouseover</b> on items shows variant details, clinical interpretation, and associated conditions. 
Further information on each variant is displayed on the details page by a click onto any variant.
</p>

<p>
Entries in the <b>ClinVar CNV track</b> are colored by <b>type of variant</b>:
<b><font color="da2c37">red for loss</font></b>,
<b><font color="294eae">blue for gain</font></b>, and
<b><font color="840876">purple for insertion</font></b>.
A light to dark color gradient indicates the clinical significance of each variant, with the 
lightest shade being benign, or others, to the darkest shade being pathogenic.
</p>

<p>
Entries in the <b>ClinVar SNVs track</b> are colored by <b>clinical significance</b>:
<b><font color="d20000">red for pathogenic</font></b>,
<b><font color="000088">dark blue for variant of uncertain significance</font></b>,
<B><font color="#00d200">green for benign</font></b>,
<B><font color="#888">dark grey for not provided</font></b>, and
<B><font color="#8979D4">light blue for conflicting</font></b>.
</p>

<p>
Items can be filtered according to the size of the variant, variant type, clinical significance, 
allele origin, and molecular consequence, using the track <b>Configure</b> options. 
Each subtrack has separate display controls, as described 
<a href="https://genome.ucsc.edu/goldenPath/help/multiView.html">here</a>.

<p>For the human genome version hg19: the hg19 genome released by UCSC in 2009 had a 
mitochondrial genome "chrM" that was not the same as the one later used for most
databases like ClinVar. As a result, we added the official mitochondrial genome
in 2020 as "chrMT" and all mitochondrial annotations of ClinVar and most other
databases are shown on the mitochondrial genome called "chrMT". For full description
of the issue of the mitochondrial genome in hg19, please see the 
<a target=_blank href="https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/">README file</a> 
on our download site. 
</p>


<h2>Data updates</h2>
<p>ClinVar publishes a new release on the 
<a href="https://www.ncbi.nlm.nih.gov/feed/rss.cgi?ChanKey=ClinVarNews">first Thursday every month</a> 
and this track is updated automatically at most six days 
later. The exact date of our last update is shown when you click onto any variant. 
You can find the previous versions of the track organized by month on our
downloads server in the 
<a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/archive/clinvar/" target="_blank">archive</a>
directory. To access one of these previous versions, paste the URL to one of
the older files into our "Custom tracks" box.</p>

<H2>Data access</H2>
<p>
The raw data can be explored interactively with the <a href="../cgi-bin/hgTables">Table Browser</a>
or the <a href="../cgi-bin/hgIntegrator">Data Integrator</a>. The data can be
accessed from scripts through our <a href="https://api.genome.ucsc.edu">API</a>, the track names are
"clinVarMain and "clinVarCnv".

<p>
For automated download and analysis, the genome annotation is stored in a bigBed file that
can be downloaded from
<a href="http://hgdownload.soe.ucsc.edu/gbdb/$db/bbi/" target="_blank">our download server</a>.
The files for this track are called <tt>clinVarMain.bb</tt> and <tt>clinVarCnv.bb</tt>. Individual
regions or the whole genome annotation can be obtained using our tool <tt>bigBedToBed</tt>
which can be compiled from the source code or downloaded as a precompiled
binary for your system. Instructions for downloading source code and binaries can be found
<a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">here</a>.
The tool
can also be used to obtain only features within a given range, e.g. 
<tt>bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/hg19/bbi/clinvar/clinvarMain.bb -chrom=chr21 -start=0 -end=100000000 stdout</tt></p>
</p>

<h2>Methods</h2>

<p>
ClinVar files were reformatted at UCSC to the <a href="../goldenPath/help/bigBed.html">bigBed</a> format.
The data is updated every month, one week after the ClinVar release date.
The program that performs the update is available on
<a href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg/utils/otto/clinvar/clinVarToBed"
target="_blank">Github</a>.
</p>

<h2>Credits</h2>
<p>
Thanks to NCBI for making the ClinVar data available on their FTP site as a tab-separated file.
</p>

<h2>References</h2>
<p>
Landrum MJ, Lee JM, Benson M, Brown G, Chao C, Chitipiralla S, Gu B, Hart J, Hoffman D, Hoover J
<em>et al</em>.
<a href="https://academic.oup.com/nar/article/44/D1/D862/2502702/ClinVar-public-archive-of-interpretations-of" target="_blank">
ClinVar: public archive of interpretations of clinically relevant variants</a>.
<em>Nucleic Acids Res</em>. 2016 Jan 4;44(D1):D862-8.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26582918" target="_blank">26582918</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4702865/" target="_blank">PMC4702865</a>
</p>
