<h2>Description</h2>

<p>
The "Constraint scores" container track includes several subtracks showing the results of
constraint prediction algorithms. These try to find regions of negative
selection, where variations likely have functional impact. The algorithms do
not use multi-species alignments to derive evolutionary constraint, but use
primarily human variation, usually from variants collected by gnomAD (see the
gnomAD V2 or V3 tracks on hg19 and hg38) or TOPMED (contained in our dbSNP
tracks and available as a filter). Another constraint score, gnomAD
constraint, is not part of this container but can be found in the hg38 gnomAD
track.

The algorithms covered here are:
<ol>
    <li><b><a href="https://github.com/astrazeneca-cgr-publications/jarvis" target="_blank">
    JARVIS - "Junk" Annotation genome-wide Residual Variation Intolerance Score</a></b>: 
    First scan the entire genome with a
    sliding-window approach (using a 1-nucleotide step), recording the number of
    all TOPMED variants and common variants, irrespective of their predicted effect,
    within each window, to eventually calculate a single-nucleotide resolution
    genome-wide residual variation intolerance score (gwRVIS). Then combine
    gwRVIS, primary genomic sequence context, and additional genomic
    annotations with a multi-module deep learning framework to infer
    pathogenicity of noncoding regions that still remains na√Øve to existing
    phylogenetic conservation metrics

    <li><b><a href="https://www.cardiodb.org/hmc/" target="_blank">
    HMC - Homologous Missense Constraint</a></b>:
    Homologous Missense Constraint (HMC) is a amino acid level measure
    of genetic intolerance of missense variants within human populations.
    For all assessable amino-acid positions in Pfam domains, the number of
    missense substitutions directly observed in gnomAD (Observed) was counted
    and compared to the expected value under a neutral evolution
    model (Expected). The upper limit of a 95% confidence interval for the
    Observed/Expected ratio is defined as the HMC score. Missense variants
    disrupting the amino-acid positions with HMC&lt;0.8 are predicted to be
    likely deleterious
   
    <li><b><a href="http://biosig.unimelb.edu.au/mtr-viewer/" target="_blank">
    MTR - Missense Tolerance Ratio</a> (hg19 only)</b>:
    Missense Tolerance Ratio (MTR) scores aim to quantify the amount of purifying 
    selection acting specifically on missense variants in a given window of 
    protein-coding sequence. It is estimated across sliding windows of 31 codons 
    (default) and uses observed standing variation data from the WES component of 
    gnomAD / the Exome Aggregation Consortium Database (ExAC), version 2.0. Scores
    were computed using Ensembl v95 release 
</ol>

<h2>Display Conventions and Configuration</h2>

<p>
Shown are the scores as a signal ("wiggle") track, with one score per genome position.
Mouse over the bars to see the exact values. 
</p>

<p>
For HMC, the highly-constrained cutoff 0.8 is indicated with a line.
</p>

<h3>MTR</h3>
<p>
MTR data can be found on two tracks, <b>MTR All data</b> and <b>MTR Scores</b>. In the
<b>MTR Scores</b> track the data has been converted into 4 separate signal tracks 
representing each base pair mutation, with the highest possible score represented when
multiple transcripts overlap. It is recommended that the data be explored using
this version of the track, as it condenses the information substatially while
retaining the magnitude of the data.</p>

<p>Any specific point mutations of interest can then be researched in the <b>
MTR All data</b> track. This track contains all of the information from 
<a href="http://biosig.unimelb.edu.au/mtr-viewer/downloads" target="_blank">
MTRV2</a> including more than 3 possible scores per base when transcripts overlap.
A mouse-over on this track shows the ref and alt allele, as well as the MTR score
and the MTR score percentile. Filters are available for MTR score, False Discovery Rate
(FDR), MTR percentile, and variant consequence. Items in the track are colored according
to their MTR percentile:</p>
<ul>
<li><b><font color=red>Red items</font></b> MTR percentiles over 75
<li><b><font color=black>Black items</font></b> MTR percentiles between 50 and 75
<li><b><font color=green>Green items</font></b> MTR percentiles below 25
<li><b><font color=blue>Blue items</font></b> No MTR score
</ul>

<p>
By default, only scores with an MTR percentile over 75 are shown.</p>

<h2>Methods</h2>

<h3>HMC</h3> 
<p>
Scores were downloaded and converted to .bedGraph files with a custom Python 
script. The bedGraph files were then converted to bigWig files, as documented in our 
<a href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg/makeDb/doc/hg19.txt" 
target=_blank>makeDoc</a> hg19 build log.</p>
<h3>Jarvis</h3> 
<p>
Scores were downloaded and converted to a single bigWig file. See 
<a href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg/makeDb/doc/hg19.txt" 
target=_blank>hg19 makeDoc</a> and  <a 
href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg/makeDb/doc/hg38/jarvis.txt" 
target=_blank>hg38 makeDoc</a></p>
<h3>MTR</h3> 
<p>
<a href="http://biosig.unimelb.edu.au/mtr-viewer/downloads" target="_blank">V2
file</a> was downloaded and columns were reshuffled as well as itemRgb added for the
<b>MTR All data</b> track. For the <b>MTR Scores</b> track the file was parsed with a python
script to pull out the highest possible MTR score for each of the 3 possible mutations
at each base pair and 4 tracks built out of these values representing each mutation.</p>
<p>
See the <a href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg
/makeDb/doc/hg19.txt" target=_blank>hg19 makeDoc</a> entry on MTR for more info.</p>

<h2>Credits</h2>

<p>
Thanks to Jean-Madeleine Desainteagathe (APHP Paris, France) for suggesting the Jarvis, MTR, HMC tracks. Thanks to Xialei Zhang for providing the HMC data file and to Dimitrios Vitsios for helping clean up the hg38 Jarvis files. 
</p>

<h2>References</h2>

<p>
Xiaolei Zhang, Pantazis I. Theotokis, Nicholas Li, the SHaRe Investigators, Caroline F. Wright, Kaitlin E. Samocha, Nicola Whiffin, James S. Ware
<a href="https://doi.org/10.1101/2022.02.16.22271023" target="_blank">
Genetic constraint at single amino acid resolution improves missense variant prioritisation and gene discovery</a>.
<em>Medrxiv</em> 2022.02.16.22271023
</p>
<p>
Silk M, Petrovski S, Ascher DB.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/31170280" target="_blank">
MTR-Viewer: identifying regions within genes under purifying selection</a>.
<em>Nucleic Acids Res</em>. 2019 Jul 2;47(W1):W121-W126.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31170280" target="_blank">31170280</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6602522/" target="_blank">PMC6602522</a>
</p>
<p>
Vitsios D, Dhindsa RS, Middleton L, Gussow AB, Petrovski S.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/33686085" target="_blank">
Prioritizing non-coding regions based on human genomic constraint and sequence context with deep
learning</a>.
<em>Nat Commun</em>. 2021 Mar 8;12(1):1504.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33686085" target="_blank">33686085</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7940646/" target="_blank">PMC7940646</a>
</p>
