<h2>Description</h2>
<p>
This track shows variants associated with monogenic congenital defects of immunity to <b>SARS-CoV-2</b>, 
with incomplete or complete penetrance from the <a target=_blank 
href="https://www.covidhge.com/">COVID Human Genetic Effort</a>. This international consortium aims 
to detect rare or common monogenic inborn immunity errors (IEI) underlying severe forms of COVID-19 
in previously healthy individuals, as well as rare or common monogenic variations that make certain 
individuals resistant to the infection by SARS-CoV2 itself despite repeated exposure.
</p>

<p>
The major feature of the small set of  variants in this track is that they are functionally tested 
to be <b>deleterious</b> and genetically tested to be <b>disease-causing</b>. Specifically, 
rare variants were predicted to be loss-of-function (LOF) at 13 human loci known to govern 
TLR3- and IRF7-dependent type I interferon (IFN) immunity to influenza virus in patients with 
life-threatening COVID-19 pneumonia, relative to subjects with asymptomatic or benign infection. 
These genetic defects display incomplete penetrance for influenza respiratory distress and only 
manifested clinically upon infection with the more virulent SARS-CoV-2.
</p>
 
<h2>Display Conventions</h2>



<h2>Methods</h2>
<p>
Variant calls in 12 autosomal IFN-related genes from whole exome or genome data with a MAF lower 
than 0.001 (gnomAD v2.1.1) and experimental demonstration of loss-of-function (LOF) were considered 
for statistical analysis. Proportion of individuals carrying at least one pLOF variant was compared 
between cases and controls by means of logistic regression with the likelihood ratio test. The first 
three principal components of the PCA were included in the logistic regression model to account for 
ethnic heterogeneity of the cohorts. Analysis of enrichment in rare (MAF &lt 0.001) synonymous 
variants of the 12 genes was performed to check the calibration of the burden test. PCA war 
conducted with Plink v1.9 software on whole exome and genome sequencing data and 1000 Genomes (1kG) 
Project phase 3 public database as reference, using 27,480 exonic variants with a minor allele 
frequency &gt0.01 and a call rate &gt0.99. The odds ratio was also estimated by logistic regression 
and adjusted for ethnic heterogeneity. 
</p>

<h2>Data Access</h2>
<p>
The raw data can be explored interactively with the <a target="_blank" href="../cgi-bin/hgTables">
Table Browser</a>, or the <a target="_blank" href="../cgi-bin/hgIntegrator">Data Integrator</a>.
Please refer to
our <a href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome"
target="_blank">mailing list archives</a> for questions, or our <a target="_blank"
href="../FAQ/FAQdownloads.html#download36">Data Access FAQ</a> for more information.
</p>

<h2>Credits</h2>
<p>
Thanks to the COVID Human Genetic Effort contributors for making these data available, and in 
particular to Qian Zhang at the Rockefeller University for review and input during browser track 
development.
</p>


<h2>References</h2>
<p>
Zhang Q, Bastard P, Liu Z, Le Pen J, Moncada-Velez M, Chen J, Ogishi M, Sabli IKD, Hodeib S, Korol C
<em>et al</em>.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/32972995" target="_blank">
Inborn errors of type I IFN immunity in patients with life-threatening COVID-19</a>.
<em>Science</em>. 2020 Sep 24;.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32972995" target="_blank">32972995</a>
</p> 

    <h2>

