<h2>Display Conventions</h2>
<p>
Unlike a regular genome browser track, the <b>COVID-19 Immunity Variants</b> track displays all
affected genes in <a href="https://genome.ucsc.edu/goldenPath/help/multiRegionHelp.html">multi-region view</a>,
showing the variants within the coordinates of the gene, but hiding all the intergenic bases. To
return to the default Genome Browser view, click on the "-" button next to the "region" box.
</p>
<p>
Only eight genes and few variants are contained in this track. Use the links in the table to
browse the gene of interest:
</p>

<table class="stdTbl">
  <tr>
    <th>Gene Name</th>
    <th>Human GRCh37/hg19 Assembly</th>
    <th>Human GRCh38/hg38 Assembly</th>
  </tr>
  <tr>
   <td>TLR3</td>
    <td><a href="../cgi-bin/hgTracks?db=hg19&covidHgeMuts=pack&position=chr4:186990309-187006252">
chr4:186990309-187006252</a></td>
    <td><a href="../cgi-bin/hgTracks?db=hg38&covidHgeMuts=pack&position=chr4:186069152-186088069">
chr4:186069152-186088069</a></td>
  </tr>
  <tr>
    <td>IRF7</td>
    <td><a href="../cgi-bin/hgTracks?db=hg19&covidHgeMuts=pack&position=chr11:612555-615999">
chr11:612555-615999</a></td>
    <td><a href="../cgi-bin/hgTracks?db=hg38&covidHgeMuts=pack&position=chr11:612591-615970">
chr11:612591-615970</a></td>
  </tr>
  <tr>
    <td>UNC93B1</td>
    <td><a href="../cgi-bin/hgTracks?db=hg19&covidHgeMuts=pack&position=chr11:67758575-67771593">
chr11:67758575-67771593</a></td>
    <td><a href="../cgi-bin/hgTracks?db=hg38&covidHgeMuts=pack&position=chr11:67991100-68004097">
chr11:67991100-68004097</a></td>
  </tr>
  <tr>
    <td>TBK1</td>
    <td><a href="../cgi-bin/hgTracks?db=hg19&covidHgeMuts=pack&position=chr12:64845840-64895899">
chr12:64845840-64895899</a></td>
    <td><a href="../cgi-bin/hgTracks?db=hg38&covidHgeMuts=pack&position=chr12:64452120-64502114">
chr12:64452120-64502114</a></td>
  </tr>
  <tr>
    <td>TICAM1</td>
    <td><a href="../cgi-bin/hgTracks?db=hg19&covidHgeMuts=pack&position=chr19:4815936-4831754">
chr19:4815936-4831754</a></td>
    <td><a href="../cgi-bin/hgTracks?db=hg38&covidHgeMuts=pack&position=chr19:4815932-4831704">
chr19:4815932-4831704</a></td>
  </tr>
  <tr>
    <td>IRF3</td>
    <td><a href="../cgi-bin/hgTracks?db=hg19&covidHgeMuts=pack&position=chr19:50162826-50169132">
chr19:50162826-50169132</a></td>
    <td><a href="../cgi-bin/hgTracks?db=hg38&covidHgeMuts=pack&position=chr19:49659570-49665875">
chr19:49659570-49665875</a></td>
  </tr>
  <tr>
    <td>IFNAR1</td>
   <td><a href="../cgi-bin/hgTracks?db=hg19&covidHgeMuts=pack&position=chr21:34697214-34732128">
chr21:34697214-34732128</a></td>
    <td><a href="../cgi-bin/hgTracks?db=hg38&covidHgeMuts=pack&position=chr21:33324970-33359864">
chr21:33324970-33359864</a></td>
  </tr>
  <tr>
    <td>IFNAR2</td>
    <td><a href="../cgi-bin/hgTracks?db=hg19&covidHgeMuts=pack&position=chr21:34602231-34636820">
chr21:34602231-34636820</a></td>
    <td><a href="../cgi-bin/hgTracks?db=hg38&covidHgeMuts=pack&position=chr21:33229974-33264525">
chr21:33229974-33264525</a></td>
  </tr>
</table>

<h2>Methods</h2>
<p>
Variant calls in 12 autosomal IFN-related genes from whole exome or genome data with a MAF lower
than 0.001 (gnomAD v2.1.1) and experimental demonstration of loss-of-function (LOF) were considered
for statistical analysis. The proportion of individuals carrying at least one pLOF variant was
compared between cases and controls by means of logistic regression with the likelihood ratio test.
The first three principal components of the PCA were included in the logistic regression model
to account for ethnic heterogeneity of the cohorts.
Analysis of enrichment in rare (MAF &lt 0.001) synonymous
variants of the 12 genes was performed to check the calibration of the burden test. PCA was
conducted with Plink v1.9 software on whole exome and genome sequencing data and 1000 Genomes (1kG)
Project phase 3 public database as reference, using 27,480 exonic variants with a minor allele
frequency &gt0.01 and a call rate &gt0.99. The odds ratio was also estimated by logistic regression
and adjusted for ethnic heterogeneity.
</p>

<h2>Data Access</h2>
<p>
The raw data can be explored interactively with the <a target="_blank" href="../cgi-bin/hgTables">
Table Browser</a>, or the <a target="_blank" href="../cgi-bin/hgIntegrator">Data Integrator</a>.
Please refer to
our <a href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome"
target="_blank">mailing list archives</a> for questions, or our <a target="_blank"
href="../FAQ/FAQdownloads.html#download36">Data Access FAQ</a> for more information.
</p>

<h2>Credits</h2>
<p>
Thanks to the COVID Human Genetic Effort contributors for making these data available, and in
particular to Qian Zhang at the Rockefeller University for review and input during browser track
development.
</p>

<h2>References</h2>
<p>
Zhang Q, Bastard P, Liu Z, Le Pen J, Moncada-Velez M, Chen J, Ogishi M, Sabli IKD, Hodeib S, Korol C
<em>et al</em>.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/32972995" target="_blank">
Inborn errors of type I IFN immunity in patients with life-threatening COVID-19</a>.
<em>Science</em>. 2020 Sep 24;.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32972995" target="_blank">32972995</a>
</p>

