<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser FAQ" -->
<!--#set var="ROOT" value=".." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>Frequently Asked Questions: Gene tracks</h1>

<h2>Topics</h2>

<ul>
<li><a href="#gene">What is a gene?</a></li>
<li><a href="#genestrans">What is the difference between a gene and a transcript?</a></li>
<li><a href="#mostCommon">What are the most common gene transcript tracks?</a></li>
<li><a href="#genename">What is a gene name?</a></li>
<li><a href="#ens">What are Ensembl and GENCODE and is there a difference?</a></li>
<li><a href="#ensRefseq">What are the differences among GENCODE, Ensembl and RefSeq?</a></li>
<li><a href="#hg19">For the human assembly hg19/GRCh37: What is the difference between UCSC 
                    Genes track, the "GENCODE" track and the "Ensembl Genes" track?</a></li>
<li><a href="#hg38">For the human assembly hg38/GRCh38: What are the differences between the 
		    "GENCODE" and "All GENCODE" tracks?</a></li>
<li><a href="#gencode">What is the difference between GENCODE comprehensive and basic?</a></li>
<li><a href="#ncbiRefseq">What is the difference between "NCBI RefSeq" and "UCSC RefSeq"?</a></li>
<li><a href="#ccds">What is CCDS?</a></li>
<li><a href="#justsingle">How can I just show a single transcript per gene?</a></li>
<li><a href="#whatdo">This is rather complicated. Can you tell me which gene transcript track
                      I should use?</a></li>
</ul>
<hr>
<p>
<a href="index.html">Return to FAQ Table of Contents</a></p>

<a name="gene"></a>
<h2>The basics</h2>
<h6>What is a gene?</h6>
<p>
Before DNA sequencing, genes were defined as heritable traits. In the present day context of 
bioinformatics, a gene represents a collection of transcripts usually transcribed within certain 
genomic coordinates . Transcripts either encode one protein or are non-coding. 
For human, most genes have an associated symbol assigned by the <a target=_blank 
href="https://www.genenames.org/">Human Gene Nomenclature Commitee (HGNC, formerly HUGO)</a>. 
For other organisms there is usually a database curation 
team that assigns symbols, such as MGI for mouse.
</p>

<a name="genestrans"></a>
<h6>What is the difference between a gene and a transcript? </h6>
<p>
Transcripts are defined as RNA molecules that are copied from the DNA template of a gene. Every gene 
is comprised of a set of transcripts. In the Genome Browser, data tracks are often called 
"Genes", e.g. "Ensembl Genes" or "UCSC Genes", but they really represent transcripts 
on an assembly. Every transcript has an accession number, a sequence, and a list of exon
chrom/start/end coordinates on a genome assembly. These transcript accession numbers 
are assigned to genes. </p>

<a name="mostCommon"></a>
<h6>What are the most common gene transcript tracks?</h6>
<p>
Originally, researchers sequenced cDNA sequences, submitted them to Genbank. The
Genome Browser shows these sequences in the Genbank or the EST track (if the cDNA is just
a single read from the 5' or 3' end). From the alignment of the cDNAs and ESTs, 
the NCBI RefSeq group manually creates a smaller set of representative transcripts.
Automated programs like UCSC's or Ensembl's gene build software do the same, just
in software, which is more systematic but more error-prone.
With the arrival of GENCODE, Ensembl added a manual curation to their
human and mouse transcripts. NCBI has since also added an automated predictions pipeline with 
their tool Gnomon and its resulting "Refseq Predicted" transcripts.</p>

<p>There are many other tracks in the group "Genes and Gene Predictions".
<a target=_blank href="../cgi-bin/hgTrackUi?db=hg38&g=genscan">Genscan</a> and <a target=_blank 
href="../cgi-bin/hgTrackUi?db=hg19&g=nscanGene">N-Scan</a> are older 
transcript predictor algorithms that are based on the genome sequence alone. 
<a target=_blank href="../cgi-bin/hgTrackUi?db=hg38&g=augustusGene">Augustus</a> and <a 
target=_blank href="../cgi-bin/hgTrackUi?db=hg19&g=acembly">AceView</a> are automated 
gene-predictors that use cDNA and EST data. These and similar gene
tracks are only relevant when you are working on a particular locus where you
think that the manually curated gene models (Ensembl and RefSeq) have
errors.</p>

<a name="genename"></a>
<h6>What is a gene name? </h6>

<p>
The most common gene names (sometimes called accession numbers) encountered by users 
are either from Ensembl, GENCODE, RefSeq, or 
gene symbols. For gene symbols, such as DDX11L1, see the above question <a 
href="FAQgenes.html#gene">"What is a gene?"</a>. Ensembl/GENCODE accession numbers in 
the human genome start with ENST followed by a number, e.g. "ENST00000619216.1". Every transcript 
is assigned to a gene with identifiers that  start with ENSG and every ENSG has at least 
one ENST assigned to it. Manually curated RefSeq transcript identifiers start with 
NM_ (coding) or NR_ (noncoding), followed by a number, or XM_ if they are predicted by 
software, e.g. "NR_046018.2". A table of all RefSeq prefixes can be found on the <a target=_blank href=
"https://www.ncbi.nlm.nih.gov/books/NBK21091/table/ch18.T.refseq_accession_numbers_and_mole/?report=objectonly">
NCBI website</a>.
</p>

<a name="ens"></a>
<h2>The differences</h2>
<h6>What are Ensembl and GENCODE and is there a difference?</h6>

<p style="color:red">TODO Lou or Max: compare the Ensembl GFF with the GENCODE GFF.</p>

<p> 
Officially, the Ensembl and GENCODE gene models are the same. On the latest human and mouse genome 
assemblies (hg38 and mm10), the identifiers, transcript sequences, and exon coordinates are almost
identical between equivalent Ensembl and GENCODE versions (excluding <a target=_blank 
href="FAQdownloads.html#downloadAlt">alternative sequences</a> or <a target=_blank 
href="FAQdownloads.html#downloadFix">fix sequences</a>).</p>

<p>GENCODE uses the UCSC convention of prefixing chromosome names with "chr", e.g. "chr1" and 
"chrM", but Ensembl calls these "1" or "MT". At the time of writing (Ensembl 89), a few transcripts 
differ due to conversion issues. In addition, around 160 PAR genes are duplicated in GENCODE
but only once in GENCODE (<b>should be ensembl here?</b>). The differences affect 
fewer than 1% of the transcripts. Apart from gene annotation itself, the links to external databases
differ.</p> 

<p>The <a target=_blank
href="https://www.gencodegenes.org/human/releases.html">GENCODE Release
History</a> shows the release dates and can be linked to corresponding Ensembl
releases.  You can download the gene transcript models from the website
<a target=_blank href=https://gencodegenes.org>https://gencodegenes.org</a> or from
<a target=_blank href=http://ensembl.org>http://ensembl.org</a>. 
Foembl many applications, the files distributed on the GENCODE website
should be easier to use, as the third party database links are easier to parse
and the sequence identifiers match the UCSC genome files, at least for the
primary chromosomes.</p>

<a name="ensRefseq"></a>
<h6>What are the differences among Ensembl, GENCODE and RefSeq?</h6>
<p> 
Different institutions have different rules on how they annotate genes. E.g.
RefSeq's criteria are more stringent, so there are fewer RefSeq
transcripts than Ensembl/GENCODE transcripts. Also, RefSeq transcripts have their own
sequences independent of the genome assembly, so certain population-specific variants
may be in RefSeq that are entirely missing from the reference genome sequence. 
This has the important implication that the position of genome variants 
are harder to map to RefSeq transcripts than for GENCODE since RefSeq transcripts
can have additional sequence or missing sequence relative to the genome.</p> 

<p>The links from either transcript model to other gene-related databases are
different. In general, it seems that high-throuput sequencing data results,
e.g. RNA-seq, are often using Ensembl/GENCODE annotations and human genetics
results are reported using RefSeq annotations. It depends on your particular
project which gene model set you want to use. Over time, the two transcript
databases have been and are becoming more similar.

<a name="hg19"></a>
<h6>For the human assembly hg19/GRCh37: What is the difference between UCSC
Genes, the "GENECODE Gene Annotation" track and the "Ensembl Genes" track?</h6>
<p>The "<a target=_blank href="../cgi-bin/hgTrackUi?db=hg19&g=knownGene">UCSC Genes</a>" track, 
also called "Known Genes", is available only on
assemblies before hg38.  It was built with a gene predictor developed at UCSC.
This gene predictor uses protein, EST and cDNA annotations to derive a
relatively restricted gene transcript set. The software is no longer in use and
there are no plans to release the track on newer human assemblies. It was last used for the 
mm10 mouse assembly. We are condering updating the hg19 annotation produced by this 
software and are interested in any user feedback on the topic.</p>

<p>The "<a target=_blank href="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeSuper">
GENECODE Gene Annotation</a>" track contains data from all versions of GENCODE. "<a target=_blank 
href="../cgi-bin/hgTrackUi?db=hg19&g=ensGene">Ensembl Genes</a>" track contains just
a single Ensembl version. See the previous question for the differences between Ensembl and GENCODE.
</p>

<a name="hg38"></a>
<h6>For the human assembly hg38/GRCh38: What are the differences between the "GENCODE" and 
"All GENCODE" tracks?</h6>
<p> 
"<a target=_blank href="../cgi-bin/hgTrackUi?db=hg38&g=knownGene">GENCODE</a>" is the default 
gene track on hg38 (similar to "Known Genes" on hg19), which means that it is associated with a 
large amount of third party information when you click on a gene. This related information is also 
avilable using the Table Browser. "<a target=_blank 
href="../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeSuper">All GENCODE</a>" contains all versions 
of GENCODE, but less third-party information.
</p>

<a name="gencode"></a>
<h6>What is the difference between "GENCODE Comprehensive" and "GENCODE Basic"?</h6>
<p> 
GENCODE Basic is the subset of GENCODE useful to most users and it is what the UCSC Genome
Browser shows by default. The "basic" tag is defined as follows in the 
<a target=_blank href="https://www.gencodegenes.org/pages/tags.html">GENCODE FAQ</a>:
"Identifies a subset of representative transcripts for each gene; prioritises
full-length protein coding transcripts over partial or non-protein coding
transcripts within the same gene, and intends to highlight those transcripts
that will be useful to the majority of users.". </p>

<a name="ncbiRefseq"></a>
<h6>What is the difference between "NCBI RefSeq" and "UCSC RefSeq"?</h6>
<p>
RefSeq gene transcripts, unlike GENCODE/Ensembl/UCSC Genes, are sequences that can differ from 
the genome. They need to be aligned to the genome to create transcript models.
Traditionally, UCSC has aligned RefSeq with BLAT (UCSC RefSeq sub-track) and NCBI has aligned with 
splign. The advantages of the UCSC alignments are that
they are updated more frequently and are available for older assemblies (like
GRCh37/hg19), but they are less stable and they are not the official alignments.
Therefore we recommend working with the NCBI annotations.
When an assembly has an "NCBI RefSeq" track, we show it by default and hide the
"UCSC RefSeq" track.
</p>

<p>
In some rare cases, the NCBI and UCSC exon boundaries differ.
Activating both RefSeq and UCSC RefSeq tracks helps you investigate the differences.
Activating the RefSeq Alignments track shows NCBI's splign alignments in more detail,
including double lines where both transcript and genomic sequence are skipped in the alignment.
When available, the RefSeq Diffs subtrack may be helpful too.
</p>

<p>
An anecdotal and rare example is SHANK2 and SHANK3 in hg19. It is impossible
for either NCBI or BLAT to get the correct alignment and gene model because the genome sequence is
missing for part of the gene.  NCBI and BLAT find slightly different exon
boundaries at the edge of the problematic region. NCBI's aligner tries very hard
to find exons that align to any transcript sequence,
so it calls a few small dubious "exons" in the affected genomic region.
GENCODE V19 also used an aligner that tried very hard to find exons, but it
found small dubious "exons" in different places than NCBI.
The RefSeq Alignments track makes the problematic region very clear with double lines
indicating unalignable transcript sequence.
</p>

<p> 
When reporting results as RefSeq coordinates, e.g. as HGVS, in research
articles, please specify the RefSeq annotation release and also the 
RefSeq transcript ID with version (e.g. NM_012309.4 not NM_012309).
Different RefSeq transcript versions have different sequence (for example,
more sequence may be added to the UTRs or even the CDS), and so the transcript coordinates
often change from one version to the next.
</p>

<a name="ccds"></a>
<h6>What is CCDS?</h6>
<p>
The <a target=_blank href="https://www.ncbi.nlm.nih.gov/projects/CCDS/CcdsBrowse.cgi">
Consensus Coding Sequence Project</a> is a list of coding sequences that are
shared between RefSeq, Ensembl/GENCODE and UCSC (hg19 "Known Genes", hg38 CCDS 
was generated with just NCBI and GENCODE) gene transcripts. You can consider 
these a subset of any gene track, filtered for high quality. The CCDS identifiers 
are very stable and allow you to link easily between the different databases. As 
the name implies, it does not cover non-coding transcripts.
</p>

<a name="justsingle"></a>
<h6>How can I just show a single transcript per gene?</h6>
<p> 
For the tracks "<a target=_blank 
href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=knownGene">
UCSC Genes</a>" (hg19) or "<a target=_blank 
href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg38&g=knownGene">
GENCODE Genes</a>" (hg38), click on their title and on the configuration page, uncheck the 
box "Show splice variants". Only a single transcript will be shown. The method how this
transcript is selected is described in the track documentation below the 
configuration settings.</p>

<p>For the track <a target=_blank 
href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg38&g=refSeqComposite">
NCBI RefSeq</a> (hg38), you can activate the subtrack "RefSeq Clinical".
It contains only the transcripts that are part of the Human Gene Mutation Database.
</p>

<a name="whatdo"></a>
<h6>This is rather complicated. Can you tell me which gene transcript track I should use?</h6>
<p> 
For automated analysis, if you are doing NGS analysis and you need to capture
all possible transcripts, GENCODE provides a comprehensive gene set.  For human 
genetics or variant annotation, a more restricted transcript set is usually sufficient and "NCBI
RefSeq" is the standard. If you are only interested in protein coding
annotations, CCDS or UniProt may be an option, but this is rather unusual.
</p>

<p>
For manual inspection of exon boundaries of a single gene, and especially if it
is a transcript that is repetitive or hard to align (e.g. very small exons),
look at the UCSC RefSeq track and watch for differences between the NCBI
and UCSC exon placement. You can also BLAT the transcript sequence. 
Manually look at ESTs, mRNAs, TransMap and possibly Augustus, Genscan, SIB, SGP
or GeneId in obscure cases where you are looking for hints on what an
alternative splicing could look like.
</p>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->

