<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser FAQ" -->
<!--#set var="ROOT" value=".." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>Frequently Asked Questions: Gene tracks</h1>

<h2>Topics</h2>

<ul>
<li><a href="#genestrans">What is the difference between genes and transcripts ?</a></li>
<li><a href="#ensgencode">What is the difference between Ensembl and Gencode ?</a></li>
<li><a href="#hg19">For the human assembly hg19/GrCH37: What is the difference between UCSC Genes, the "Gencode All" track and the "Ensembl Genes" track ?</a></li>
<li><a href="#hg38">For the human assembly hg38/GrCH38: What is the difference between "Gencode" and "All Gencode" ?</a></li>
<li><a href="#ensRefseq">What is the difference between Gencode/Ensembl and RefSeq ?</a></li>
<li><a href="#gencode">What is the difference between Gencode comprehensive and basic?</a></li>
<li><a href="#ccds">What is CCDS?</a></li>
<li><a href="#justsingle">There are too many transcripts. Can I show just a single transcript?</a></li>
<li><a href="#whatdo">This is rather complicated. Can you tell me which gene transcript track I should use?</a></li>
</ul>
<hr>
<p>
<a href="index.html">Return to FAQ Table of Contents</a></p>

<a name="genestrans"></a>
<h2>The basics</h2>
<h6>What is the difference between genes and transcripts? </h6>
<p>
There are a few general things that are true of any gene track.
Every gene has a set of transcripts. Tracks are often called "Genes", e.g.
"Ensembl Genes" or "UCSC Genes", but they really are transcripts on an assembly.
Every transcript has an
accession number, a sequence and, in the genome browser, also a list of exon
chrom/start/end coordinates on a genome assembly. These transcript accessions
are assigned to genes, so every transcript is assigned to one gene. Most of the
genes have a symbol associated to them by the Human Gene Nomenclature Commitee
(HGNC, formerly HUGO).</p>

<h6>What are common gene tracks?</h6>
<p>
Originally, researchers sequenced cDNA sequences, submitted them to Genbank and the
Genome Browser shows these in the Genbank track, or in the EST track, if the cDNA is just
a single read from the 5' or 3' end. From the alignment of the cDNAs and ESTs, 
the NCBI RefSeq group would manually create a smaller set of representative transcripts.
Automated programs like UCSC's or Ensembl's gene build software did the same, just
in software, which is more systematic but also more error-prone.
With the arrival of Gencode, Ensembl added a manual curation of their
transcripts. NCBI added with Gnomon and its "Refseq Predicted" transcripts an
automated gene build pipeline. ?Aceview? ?UCSC vs Ensembl?  </p>

<p>
Ensembl/Gencode accession numbers in the human genome start with ENST followed by a
number. The genes have identifiers that start with ENSG and every ENSG has at
least one ENST assigned to it. Manually curated Refseq transcript identifiers
start with NM_ (coding) or NR_ (noncoding), followed by a number, or XM_ if
they are predicted.  </p>

<a name="ensgencode"></a>
<h2>The differences</h2>
<h6>What is the difference between Ensembl and Gencode?</h6>
<p> 
On the human and mouse genome, the identifiers, transcript sequences and exon coordinates of
Ensembl and Gencode versions are identical. You can download the gene models either from the 
website gencodegenes.org or from http://ensembl.org. Additional information that are not directly 
transcript related, e.g. the linkouts to other databases, sometimes differ in the files that are 
distributed by Gencode and the gene model files you can find on the Ensembl website.

<a name="hg19"></a>
<h6>For the human assembly hg19/GrCH37: What is the difference between UCSC
Genes, the "Gencode All" track and the "Ensembl Genes" track ?</h6>
<p> 
UCSC vs others is easy, but what is the difference between Gencode All vs Ensembl ?
</p>

<a name="hg38"></a>
<h6>For the human assembly hg38/GrCH38: What is the difference between "Gencode" and "All Gencode" ?</h6>
<p> 
I don't really know. Is it only the meta data?
</p>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->

