
<h2>Description</h2>
<p>
The ENCODE project has established an epigenomic resource for mammalian development,
profiling a diverse panel of mouse tissues at eight developmental stages from 10.5 days post
conception until birth.</p>
<div><img border="1" src="../images/renMouseDev.png" style="margin-left: 100px;"></div>
<p>
This track set presents enhancer-promoter interactions
predicted from enhancer-associated chromatin signals and gene expression across tissue stages,
based on histone modification (ChIP-seq), chromatin accessibility (ATAC-seq) 
and transcription (RNA-seq) assays and analysis performed by the ENCODE project.</p>
<p>
The promoter annotation in this track was obtained from the 
<a target="_blank" href="https://epd.epfl.ch/EPDnew_database.php">
Eukaryotic Promoter Database</a> (EPDNew).

<h2> Display Conventions and Configuration </h2>
<p>

This track is a multi-view composite track containing four data types
(<em>views</em>):
<ul>
<li>Enhancer elements</li>
<li>Promoter regions</li>
<li>Interactions between enhancers and gene promoter</li>
<li>Clustered interactions, by promoter or enhancer</li>
</ul>
<p>
For each view, there is a choice of two subtracks to display individually on the browser;
the full set of interactions (and component promoters and enhancers), and
a more stringent set, containing only interactions replicated in two biological
replicates.
Instructions for configuring multi-view
tracks are <a href="/goldenPath/help/multiView.html" target=_blank>here</a>.</p>
<p>
The display aims to improve distinguishing genes and their interactions by coloring
successive genes in different colors.  All interactions with a gene are colored the
same as the gene.</p>

<h2>Methods</h2>
<p>
Strong enhancer calls from ChromHMM were merged and filtered to produce a set 
of 66,556 putative strong TSS-distal enhancers across all tissues and developmental stages, 
and H3K27ac signals at these regions were quantified using uniquely aligned, 
de-duplicated ChIP-seq reads.</p>
<p>
Topologically associated domains (TAD's) identified in mouse ES-cells (Dixon et al.) were 
then used to constrain enhancer/gene associations.
All protein-coding genes and putative strong enhancers within each TAD were evaluated in 
terms of the Spearman's correlation (SCC) between the H3K27ac pattern of enrichment and 
the mRNA expression from RNA-seq across samples.
The SCC of each enhancer was calculated with reference to all genes on the chromosome, and this metric was used to estimate p-value 
by two strategies.
Putative enhancers showing a p-value &lt;= 0.05 (from both strategies)and a SCC &gt;= 0.25 
were retained.
</p>
<p>
Two maps were independently derived from two biological replicates of ChIP-seq and RNA-seq data.
The "All" dataset contains significant interactions present in either biological replicate. 
The "Replicated" dataset contains significant interactions present in both
</p>
<p>
Detailed methods can be found in Gorkin et al. An atlas of dynamic chromatin landscapes in the developing mouse fetus. Nature, In Press. (pre-print: doi: https://doi.org/10.1101/166652)
<p>

<h2>Data Access</h2>
Histone ChIP-seq, RNA-seq, and ATAC-seq data underlying this annotation are available from the
ENCODE portal.

https://www.encodeproject.org/search/?type=OrganismDevelopmentSeries&internal_tags=MouseDevSeries&status=released&assembly=mm10&related_datasets.assay_term_name=RNA-seq&lab.title=Barbara+Wold%2C+Caltech

<h2>Credits</h2>

David Gorkin
Iros Barozzi
Yanxiao Zhang
Yuan Zhao

<h2>References</h2>
<p>
An atlas of dynamic chromatin landscapes in the developing mouse fetus
Gorkin et al.  In Press, Nature. 
(pre-print: doi: 
<a target="_blank">https://doi.org/10.1101/166652)</p>
</p>

<p>
Dixon JR, Selvaraj S, Yue F, Kim A, Li Y, Shen Y, Hu M, Liu JS, Ren B.
<a href="https://doi.org/10.1038/nature11082" target="_blank">
Topological domains in mammalian genomes identified by analysis of chromatin interactions</a>.
<em>Nature</em>. 2012 Apr 11;485(7398):376-80.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22495300" target="_blank">22495300</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3356448/" target="_blank">PMC3356448</a>
</p>

